// react-xml-viewer@2.0.1/dist/index.es.js
import * as qt from "react";
import Yr, { createContext as ba, useContext as ya, useMemo as qr, useState as Hr, useEffect as zr } from "react";
var $e = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function et(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var We = { exports: {} }, ie = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ht;
function $a() {
  if (Ht)
    return ie;
  Ht = 1;
  var e = Yr, t = Symbol.for("react.element"), r = Symbol.for("react.fragment"), n = Object.prototype.hasOwnProperty, a = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, s = { key: !0, ref: !0, __self: !0, __source: !0 };
  function l(u, o, c) {
    var d, g = {}, h = null, v = null;
    c !== void 0 && (h = "" + c), o.key !== void 0 && (h = "" + o.key), o.ref !== void 0 && (v = o.ref);
    for (d in o)
      n.call(o, d) && !s.hasOwnProperty(d) && (g[d] = o[d]);
    if (u && u.defaultProps)
      for (d in o = u.defaultProps, o)
        g[d] === void 0 && (g[d] = o[d]);
    return { $$typeof: t, type: u, key: h, ref: v, props: g, _owner: a.current };
  }
  return ie.Fragment = r, ie.jsx = l, ie.jsxs = l, ie;
}
var oe = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var zt;
function _a() {
  return zt || (zt = 1, process.env.NODE_ENV !== "production" && function() {
    var e = Yr, t = Symbol.for("react.element"), r = Symbol.for("react.portal"), n = Symbol.for("react.fragment"), a = Symbol.for("react.strict_mode"), s = Symbol.for("react.profiler"), l = Symbol.for("react.provider"), u = Symbol.for("react.context"), o = Symbol.for("react.forward_ref"), c = Symbol.for("react.suspense"), d = Symbol.for("react.suspense_list"), g = Symbol.for("react.memo"), h = Symbol.for("react.lazy"), v = Symbol.for("react.offscreen"), T = Symbol.iterator, w = "@@iterator";
    function C(i) {
      if (i === null || typeof i != "object")
        return null;
      var f = T && i[T] || i[w];
      return typeof f == "function" ? f : null;
    }
    var P = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function N(i) {
      {
        for (var f = arguments.length, p = new Array(f > 1 ? f - 1 : 0), y = 1; y < f; y++)
          p[y - 1] = arguments[y];
        pe("error", i, p);
      }
    }
    function pe(i, f, p) {
      {
        var y = P.ReactDebugCurrentFrame, m = y.getStackAddendum();
        m !== "" && (f += "%s", p = p.concat([m]));
        var E = p.map(function(_) {
          return String(_);
        });
        E.unshift("Warning: " + f), Function.prototype.apply.call(console[i], console, E);
      }
    }
    var q = !1, H = !1, Un = !1, kn = !1, Kn = !1, Et;
    Et = Symbol.for("react.module.reference");
    function Xn(i) {
      return !!(typeof i == "string" || typeof i == "function" || i === n || i === s || Kn || i === a || i === c || i === d || kn || i === v || q || H || Un || typeof i == "object" && i !== null && (i.$$typeof === h || i.$$typeof === g || i.$$typeof === l || i.$$typeof === u || i.$$typeof === o || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      i.$$typeof === Et || i.getModuleId !== void 0));
    }
    function Wn(i, f, p) {
      var y = i.displayName;
      if (y)
        return y;
      var m = f.displayName || f.name || "";
      return m !== "" ? p + "(" + m + ")" : p;
    }
    function At(i) {
      return i.displayName || "Context";
    }
    function M(i) {
      if (i == null)
        return null;
      if (typeof i.tag == "number" && N("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof i == "function")
        return i.displayName || i.name || null;
      if (typeof i == "string")
        return i;
      switch (i) {
        case n:
          return "Fragment";
        case r:
          return "Portal";
        case s:
          return "Profiler";
        case a:
          return "StrictMode";
        case c:
          return "Suspense";
        case d:
          return "SuspenseList";
      }
      if (typeof i == "object")
        switch (i.$$typeof) {
          case u:
            var f = i;
            return At(f) + ".Consumer";
          case l:
            var p = i;
            return At(p._context) + ".Provider";
          case o:
            return Wn(i, i.render, "ForwardRef");
          case g:
            var y = i.displayName || null;
            return y !== null ? y : M(i.type) || "Memo";
          case h: {
            var m = i, E = m._payload, _ = m._init;
            try {
              return M(_(E));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var K = Object.assign, ae = 0, Ot, Ct, Nt, wt, Pt, St, xt;
    function jt() {
    }
    jt.__reactDisabledLog = !0;
    function Yn() {
      {
        if (ae === 0) {
          Ot = console.log, Ct = console.info, Nt = console.warn, wt = console.error, Pt = console.group, St = console.groupCollapsed, xt = console.groupEnd;
          var i = {
            configurable: !0,
            enumerable: !0,
            value: jt,
            writable: !0
          };
          Object.defineProperties(console, {
            info: i,
            log: i,
            warn: i,
            error: i,
            group: i,
            groupCollapsed: i,
            groupEnd: i
          });
        }
        ae++;
      }
    }
    function qn() {
      {
        if (ae--, ae === 0) {
          var i = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: K({}, i, {
              value: Ot
            }),
            info: K({}, i, {
              value: Ct
            }),
            warn: K({}, i, {
              value: Nt
            }),
            error: K({}, i, {
              value: wt
            }),
            group: K({}, i, {
              value: Pt
            }),
            groupCollapsed: K({}, i, {
              value: St
            }),
            groupEnd: K({}, i, {
              value: xt
            })
          });
        }
        ae < 0 && N("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var Re = P.ReactCurrentDispatcher, Fe;
    function ge(i, f, p) {
      {
        if (Fe === void 0)
          try {
            throw Error();
          } catch (m) {
            var y = m.stack.trim().match(/\n( *(at )?)/);
            Fe = y && y[1] || "";
          }
        return `
` + Fe + i;
      }
    }
    var Le = !1, he;
    {
      var Hn = typeof WeakMap == "function" ? WeakMap : Map;
      he = new Hn();
    }
    function It(i, f) {
      if (!i || Le)
        return "";
      {
        var p = he.get(i);
        if (p !== void 0)
          return p;
      }
      var y;
      Le = !0;
      var m = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var E;
      E = Re.current, Re.current = null, Yn();
      try {
        if (f) {
          var _ = function() {
            throw Error();
          };
          if (Object.defineProperty(_.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(_, []);
            } catch (B) {
              y = B;
            }
            Reflect.construct(i, [], _);
          } else {
            try {
              _.call();
            } catch (B) {
              y = B;
            }
            i.call(_.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (B) {
            y = B;
          }
          i();
        }
      } catch (B) {
        if (B && y && typeof B.stack == "string") {
          for (var $ = B.stack.split(`
`), I = y.stack.split(`
`), S = $.length - 1, x = I.length - 1; S >= 1 && x >= 0 && $[S] !== I[x]; )
            x--;
          for (; S >= 1 && x >= 0; S--, x--)
            if ($[S] !== I[x]) {
              if (S !== 1 || x !== 1)
                do
                  if (S--, x--, x < 0 || $[S] !== I[x]) {
                    var F = `
` + $[S].replace(" at new ", " at ");
                    return i.displayName && F.includes("<anonymous>") && (F = F.replace("<anonymous>", i.displayName)), typeof i == "function" && he.set(i, F), F;
                  }
                while (S >= 1 && x >= 0);
              break;
            }
        }
      } finally {
        Le = !1, Re.current = E, qn(), Error.prepareStackTrace = m;
      }
      var J = i ? i.displayName || i.name : "", Yt = J ? ge(J) : "";
      return typeof i == "function" && he.set(i, Yt), Yt;
    }
    function zn(i, f, p) {
      return It(i, !1);
    }
    function Jn(i) {
      var f = i.prototype;
      return !!(f && f.isReactComponent);
    }
    function ve(i, f, p) {
      if (i == null)
        return "";
      if (typeof i == "function")
        return It(i, Jn(i));
      if (typeof i == "string")
        return ge(i);
      switch (i) {
        case c:
          return ge("Suspense");
        case d:
          return ge("SuspenseList");
      }
      if (typeof i == "object")
        switch (i.$$typeof) {
          case o:
            return zn(i.render);
          case g:
            return ve(i.type, f, p);
          case h: {
            var y = i, m = y._payload, E = y._init;
            try {
              return ve(E(m), f, p);
            } catch {
            }
          }
        }
      return "";
    }
    var be = Object.prototype.hasOwnProperty, Rt = {}, Ft = P.ReactDebugCurrentFrame;
    function ye(i) {
      if (i) {
        var f = i._owner, p = ve(i.type, i._source, f ? f.type : null);
        Ft.setExtraStackFrame(p);
      } else
        Ft.setExtraStackFrame(null);
    }
    function Zn(i, f, p, y, m) {
      {
        var E = Function.call.bind(be);
        for (var _ in i)
          if (E(i, _)) {
            var $ = void 0;
            try {
              if (typeof i[_] != "function") {
                var I = Error((y || "React class") + ": " + p + " type `" + _ + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof i[_] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw I.name = "Invariant Violation", I;
              }
              $ = i[_](f, _, y, p, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (S) {
              $ = S;
            }
            $ && !($ instanceof Error) && (ye(m), N("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", y || "React class", p, _, typeof $), ye(null)), $ instanceof Error && !($.message in Rt) && (Rt[$.message] = !0, ye(m), N("Failed %s type: %s", p, $.message), ye(null));
          }
      }
    }
    var Qn = Array.isArray;
    function Me(i) {
      return Qn(i);
    }
    function Dn(i) {
      {
        var f = typeof Symbol == "function" && Symbol.toStringTag, p = f && i[Symbol.toStringTag] || i.constructor.name || "Object";
        return p;
      }
    }
    function ea(i) {
      try {
        return Lt(i), !1;
      } catch {
        return !0;
      }
    }
    function Lt(i) {
      return "" + i;
    }
    function Mt(i) {
      if (ea(i))
        return N("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", Dn(i)), Lt(i);
    }
    var se = P.ReactCurrentOwner, ta = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Bt, Gt, Be;
    Be = {};
    function ra(i) {
      if (be.call(i, "ref")) {
        var f = Object.getOwnPropertyDescriptor(i, "ref").get;
        if (f && f.isReactWarning)
          return !1;
      }
      return i.ref !== void 0;
    }
    function na(i) {
      if (be.call(i, "key")) {
        var f = Object.getOwnPropertyDescriptor(i, "key").get;
        if (f && f.isReactWarning)
          return !1;
      }
      return i.key !== void 0;
    }
    function aa(i, f) {
      if (typeof i.ref == "string" && se.current && f && se.current.stateNode !== f) {
        var p = M(se.current.type);
        Be[p] || (N('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', M(se.current.type), i.ref), Be[p] = !0);
      }
    }
    function sa(i, f) {
      {
        var p = function() {
          Bt || (Bt = !0, N("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", f));
        };
        p.isReactWarning = !0, Object.defineProperty(i, "key", {
          get: p,
          configurable: !0
        });
      }
    }
    function ia(i, f) {
      {
        var p = function() {
          Gt || (Gt = !0, N("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", f));
        };
        p.isReactWarning = !0, Object.defineProperty(i, "ref", {
          get: p,
          configurable: !0
        });
      }
    }
    var oa = function(i, f, p, y, m, E, _) {
      var $ = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: t,
        // Built-in properties that belong on the element
        type: i,
        key: f,
        ref: p,
        props: _,
        // Record the component responsible for creating this element.
        _owner: E
      };
      return $._store = {}, Object.defineProperty($._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty($, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: y
      }), Object.defineProperty($, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: m
      }), Object.freeze && (Object.freeze($.props), Object.freeze($)), $;
    };
    function ua(i, f, p, y, m) {
      {
        var E, _ = {}, $ = null, I = null;
        p !== void 0 && (Mt(p), $ = "" + p), na(f) && (Mt(f.key), $ = "" + f.key), ra(f) && (I = f.ref, aa(f, m));
        for (E in f)
          be.call(f, E) && !ta.hasOwnProperty(E) && (_[E] = f[E]);
        if (i && i.defaultProps) {
          var S = i.defaultProps;
          for (E in S)
            _[E] === void 0 && (_[E] = S[E]);
        }
        if ($ || I) {
          var x = typeof i == "function" ? i.displayName || i.name || "Unknown" : i;
          $ && sa(_, x), I && ia(_, x);
        }
        return oa(i, $, I, m, y, se.current, _);
      }
    }
    var Ge = P.ReactCurrentOwner, Vt = P.ReactDebugCurrentFrame;
    function z(i) {
      if (i) {
        var f = i._owner, p = ve(i.type, i._source, f ? f.type : null);
        Vt.setExtraStackFrame(p);
      } else
        Vt.setExtraStackFrame(null);
    }
    var Ve;
    Ve = !1;
    function Ue(i) {
      return typeof i == "object" && i !== null && i.$$typeof === t;
    }
    function Ut() {
      {
        if (Ge.current) {
          var i = M(Ge.current.type);
          if (i)
            return `

Check the render method of \`` + i + "`.";
        }
        return "";
      }
    }
    function la(i) {
      {
        if (i !== void 0) {
          var f = i.fileName.replace(/^.*[\\\/]/, ""), p = i.lineNumber;
          return `

Check your code at ` + f + ":" + p + ".";
        }
        return "";
      }
    }
    var kt = {};
    function ca(i) {
      {
        var f = Ut();
        if (!f) {
          var p = typeof i == "string" ? i : i.displayName || i.name;
          p && (f = `

Check the top-level render call using <` + p + ">.");
        }
        return f;
      }
    }
    function Kt(i, f) {
      {
        if (!i._store || i._store.validated || i.key != null)
          return;
        i._store.validated = !0;
        var p = ca(f);
        if (kt[p])
          return;
        kt[p] = !0;
        var y = "";
        i && i._owner && i._owner !== Ge.current && (y = " It was passed a child from " + M(i._owner.type) + "."), z(i), N('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', p, y), z(null);
      }
    }
    function Xt(i, f) {
      {
        if (typeof i != "object")
          return;
        if (Me(i))
          for (var p = 0; p < i.length; p++) {
            var y = i[p];
            Ue(y) && Kt(y, f);
          }
        else if (Ue(i))
          i._store && (i._store.validated = !0);
        else if (i) {
          var m = C(i);
          if (typeof m == "function" && m !== i.entries)
            for (var E = m.call(i), _; !(_ = E.next()).done; )
              Ue(_.value) && Kt(_.value, f);
        }
      }
    }
    function fa(i) {
      {
        var f = i.type;
        if (f == null || typeof f == "string")
          return;
        var p;
        if (typeof f == "function")
          p = f.propTypes;
        else if (typeof f == "object" && (f.$$typeof === o || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        f.$$typeof === g))
          p = f.propTypes;
        else
          return;
        if (p) {
          var y = M(f);
          Zn(p, i.props, "prop", y, i);
        } else if (f.PropTypes !== void 0 && !Ve) {
          Ve = !0;
          var m = M(f);
          N("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", m || "Unknown");
        }
        typeof f.getDefaultProps == "function" && !f.getDefaultProps.isReactClassApproved && N("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function da(i) {
      {
        for (var f = Object.keys(i.props), p = 0; p < f.length; p++) {
          var y = f[p];
          if (y !== "children" && y !== "key") {
            z(i), N("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", y), z(null);
            break;
          }
        }
        i.ref !== null && (z(i), N("Invalid attribute `ref` supplied to `React.Fragment`."), z(null));
      }
    }
    function Wt(i, f, p, y, m, E) {
      {
        var _ = Xn(i);
        if (!_) {
          var $ = "";
          (i === void 0 || typeof i == "object" && i !== null && Object.keys(i).length === 0) && ($ += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var I = la(m);
          I ? $ += I : $ += Ut();
          var S;
          i === null ? S = "null" : Me(i) ? S = "array" : i !== void 0 && i.$$typeof === t ? (S = "<" + (M(i.type) || "Unknown") + " />", $ = " Did you accidentally export a JSX literal instead of a component?") : S = typeof i, N("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", S, $);
        }
        var x = ua(i, f, p, m, E);
        if (x == null)
          return x;
        if (_) {
          var F = f.children;
          if (F !== void 0)
            if (y)
              if (Me(F)) {
                for (var J = 0; J < F.length; J++)
                  Xt(F[J], i);
                Object.freeze && Object.freeze(F);
              } else
                N("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Xt(F, i);
        }
        return i === n ? da(x) : fa(x), x;
      }
    }
    function pa(i, f, p) {
      return Wt(i, f, p, !0);
    }
    function ga(i, f, p) {
      return Wt(i, f, p, !1);
    }
    var ha = ga, va = pa;
    oe.Fragment = n, oe.jsx = ha, oe.jsxs = va;
  }()), oe;
}
process.env.NODE_ENV === "production" ? We.exports = $a() : We.exports = _a();
var b = We.exports;
const Jt = ":@", Ee = "#text", tt = "#comment", rt = "#cdata", Jr = "DECLARATION_TAG", Ta = "TAG", Ye = {
  tagColor: "#d43900",
  textColor: "#333",
  attributeKeyColor: "#2a7ab0",
  attributeValueColor: "#008000",
  separatorColor: "#333",
  commentColor: "#aaa",
  cdataColor: "#1d781d",
  fontFamily: "monospace"
}, ma = {
  theme: Ye,
  collapsible: !1,
  indentSize: 2
}, Zr = ba(ma), G = () => ya(Zr);
var nt = {}, we = {};
(function(e) {
  const t = ":A-Za-z_\\u00C0-\\u00D6\\u00D8-\\u00F6\\u00F8-\\u02FF\\u0370-\\u037D\\u037F-\\u1FFF\\u200C-\\u200D\\u2070-\\u218F\\u2C00-\\u2FEF\\u3001-\\uD7FF\\uF900-\\uFDCF\\uFDF0-\\uFFFD", r = t + "\\-.\\d\\u00B7\\u0300-\\u036F\\u203F-\\u2040", n = "[" + t + "][" + r + "]*", a = new RegExp("^" + n + "$"), s = function(u, o) {
    const c = [];
    let d = o.exec(u);
    for (; d; ) {
      const g = [];
      g.startIndex = o.lastIndex - d[0].length;
      const h = d.length;
      for (let v = 0; v < h; v++)
        g.push(d[v]);
      c.push(g), d = o.exec(u);
    }
    return c;
  }, l = function(u) {
    const o = a.exec(u);
    return !(o === null || typeof o > "u");
  };
  e.isExist = function(u) {
    return typeof u < "u";
  }, e.isEmptyObject = function(u) {
    return Object.keys(u).length === 0;
  }, e.merge = function(u, o, c) {
    if (o) {
      const d = Object.keys(o), g = d.length;
      for (let h = 0; h < g; h++)
        c === "strict" ? u[d[h]] = [o[d[h]]] : u[d[h]] = o[d[h]];
    }
  }, e.getValue = function(u) {
    return e.isExist(u) ? u : "";
  }, e.isName = l, e.getAllMatches = s, e.nameRegexp = n;
})(we);
const at = we, Ea = {
  allowBooleanAttributes: !1,
  //A tag can have attributes without any value
  unpairedTags: []
};
nt.validate = function(e, t) {
  t = Object.assign({}, Ea, t);
  const r = [];
  let n = !1, a = !1;
  e[0] === "\uFEFF" && (e = e.substr(1));
  for (let s = 0; s < e.length; s++)
    if (e[s] === "<" && e[s + 1] === "?") {
      if (s += 2, s = Qt(e, s), s.err)
        return s;
    } else if (e[s] === "<") {
      let l = s;
      if (s++, e[s] === "!") {
        s = Dt(e, s);
        continue;
      } else {
        let u = !1;
        e[s] === "/" && (u = !0, s++);
        let o = "";
        for (; s < e.length && e[s] !== ">" && e[s] !== " " && e[s] !== "	" && e[s] !== `
` && e[s] !== "\r"; s++)
          o += e[s];
        if (o = o.trim(), o[o.length - 1] === "/" && (o = o.substring(0, o.length - 1), s--), !xa(o)) {
          let g;
          return o.trim().length === 0 ? g = "Invalid space after '<'." : g = "Tag '" + o + "' is an invalid name.", j("InvalidTag", g, R(e, s));
        }
        const c = Ca(e, s);
        if (c === !1)
          return j("InvalidAttr", "Attributes for '" + o + "' have open quote.", R(e, s));
        let d = c.value;
        if (s = c.index, d[d.length - 1] === "/") {
          const g = s - d.length;
          d = d.substring(0, d.length - 1);
          const h = er(d, t);
          if (h === !0)
            n = !0;
          else
            return j(h.err.code, h.err.msg, R(e, g + h.err.line));
        } else if (u)
          if (c.tagClosed) {
            if (d.trim().length > 0)
              return j("InvalidTag", "Closing tag '" + o + "' can't have attributes or invalid starting.", R(e, l));
            {
              const g = r.pop();
              if (o !== g.tagName) {
                let h = R(e, g.tagStartPos);
                return j(
                  "InvalidTag",
                  "Expected closing tag '" + g.tagName + "' (opened in line " + h.line + ", col " + h.col + ") instead of closing tag '" + o + "'.",
                  R(e, l)
                );
              }
              r.length == 0 && (a = !0);
            }
          } else
            return j("InvalidTag", "Closing tag '" + o + "' doesn't have proper closing.", R(e, s));
        else {
          const g = er(d, t);
          if (g !== !0)
            return j(g.err.code, g.err.msg, R(e, s - d.length + g.err.line));
          if (a === !0)
            return j("InvalidXml", "Multiple possible root nodes found.", R(e, s));
          t.unpairedTags.indexOf(o) !== -1 || r.push({ tagName: o, tagStartPos: l }), n = !0;
        }
        for (s++; s < e.length; s++)
          if (e[s] === "<")
            if (e[s + 1] === "!") {
              s++, s = Dt(e, s);
              continue;
            } else if (e[s + 1] === "?") {
              if (s = Qt(e, ++s), s.err)
                return s;
            } else
              break;
          else if (e[s] === "&") {
            const g = Pa(e, s);
            if (g == -1)
              return j("InvalidChar", "char '&' is not expected.", R(e, s));
            s = g;
          } else if (a === !0 && !Zt(e[s]))
            return j("InvalidXml", "Extra text at the end", R(e, s));
        e[s] === "<" && s--;
      }
    } else {
      if (Zt(e[s]))
        continue;
      return j("InvalidChar", "char '" + e[s] + "' is not expected.", R(e, s));
    }
  if (n) {
    if (r.length == 1)
      return j("InvalidTag", "Unclosed tag '" + r[0].tagName + "'.", R(e, r[0].tagStartPos));
    if (r.length > 0)
      return j("InvalidXml", "Invalid '" + JSON.stringify(r.map((s) => s.tagName), null, 4).replace(/\r?\n/g, "") + "' found.", { line: 1, col: 1 });
  } else
    return j("InvalidXml", "Start tag expected.", 1);
  return !0;
};
function Zt(e) {
  return e === " " || e === "	" || e === `
` || e === "\r";
}
function Qt(e, t) {
  const r = t;
  for (; t < e.length; t++)
    if (e[t] == "?" || e[t] == " ") {
      const n = e.substr(r, t - r);
      if (t > 5 && n === "xml")
        return j("InvalidXml", "XML declaration allowed only at the start of the document.", R(e, t));
      if (e[t] == "?" && e[t + 1] == ">") {
        t++;
        break;
      } else
        continue;
    }
  return t;
}
function Dt(e, t) {
  if (e.length > t + 5 && e[t + 1] === "-" && e[t + 2] === "-") {
    for (t += 3; t < e.length; t++)
      if (e[t] === "-" && e[t + 1] === "-" && e[t + 2] === ">") {
        t += 2;
        break;
      }
  } else if (e.length > t + 8 && e[t + 1] === "D" && e[t + 2] === "O" && e[t + 3] === "C" && e[t + 4] === "T" && e[t + 5] === "Y" && e[t + 6] === "P" && e[t + 7] === "E") {
    let r = 1;
    for (t += 8; t < e.length; t++)
      if (e[t] === "<")
        r++;
      else if (e[t] === ">" && (r--, r === 0))
        break;
  } else if (e.length > t + 9 && e[t + 1] === "[" && e[t + 2] === "C" && e[t + 3] === "D" && e[t + 4] === "A" && e[t + 5] === "T" && e[t + 6] === "A" && e[t + 7] === "[") {
    for (t += 8; t < e.length; t++)
      if (e[t] === "]" && e[t + 1] === "]" && e[t + 2] === ">") {
        t += 2;
        break;
      }
  }
  return t;
}
const Aa = '"', Oa = "'";
function Ca(e, t) {
  let r = "", n = "", a = !1;
  for (; t < e.length; t++) {
    if (e[t] === Aa || e[t] === Oa)
      n === "" ? n = e[t] : n !== e[t] || (n = "");
    else if (e[t] === ">" && n === "") {
      a = !0;
      break;
    }
    r += e[t];
  }
  return n !== "" ? !1 : {
    value: r,
    index: t,
    tagClosed: a
  };
}
const Na = new RegExp(`(\\s*)([^\\s=]+)(\\s*=)?(\\s*(['"])(([\\s\\S])*?)\\5)?`, "g");
function er(e, t) {
  const r = at.getAllMatches(e, Na), n = {};
  for (let a = 0; a < r.length; a++) {
    if (r[a][1].length === 0)
      return j("InvalidAttr", "Attribute '" + r[a][2] + "' has no space in starting.", ue(r[a]));
    if (r[a][3] !== void 0 && r[a][4] === void 0)
      return j("InvalidAttr", "Attribute '" + r[a][2] + "' is without value.", ue(r[a]));
    if (r[a][3] === void 0 && !t.allowBooleanAttributes)
      return j("InvalidAttr", "boolean attribute '" + r[a][2] + "' is not allowed.", ue(r[a]));
    const s = r[a][2];
    if (!Sa(s))
      return j("InvalidAttr", "Attribute '" + s + "' is an invalid name.", ue(r[a]));
    if (!n.hasOwnProperty(s))
      n[s] = 1;
    else
      return j("InvalidAttr", "Attribute '" + s + "' is repeated.", ue(r[a]));
  }
  return !0;
}
function wa(e, t) {
  let r = /\d/;
  for (e[t] === "x" && (t++, r = /[\da-fA-F]/); t < e.length; t++) {
    if (e[t] === ";")
      return t;
    if (!e[t].match(r))
      break;
  }
  return -1;
}
function Pa(e, t) {
  if (t++, e[t] === ";")
    return -1;
  if (e[t] === "#")
    return t++, wa(e, t);
  let r = 0;
  for (; t < e.length; t++, r++)
    if (!(e[t].match(/\w/) && r < 20)) {
      if (e[t] === ";")
        break;
      return -1;
    }
  return t;
}
function j(e, t, r) {
  return {
    err: {
      code: e,
      msg: t,
      line: r.line || r,
      col: r.col
    }
  };
}
function Sa(e) {
  return at.isName(e);
}
function xa(e) {
  return at.isName(e);
}
function R(e, t) {
  const r = e.substring(0, t).split(/\r?\n/);
  return {
    line: r.length,
    // column number is last line's length + 1, because column numbering starts at 1:
    col: r[r.length - 1].length + 1
  };
}
function ue(e) {
  return e.startIndex + e[1].length;
}
var st = {};
const Qr = {
  preserveOrder: !1,
  attributeNamePrefix: "@_",
  attributesGroupName: !1,
  textNodeName: "#text",
  ignoreAttributes: !0,
  removeNSPrefix: !1,
  // remove NS from tag name or attribute name if true
  allowBooleanAttributes: !1,
  //a tag can have attributes without any value
  //ignoreRootElement : false,
  parseTagValue: !0,
  parseAttributeValue: !1,
  trimValues: !0,
  //Trim string values of tag and attributes
  cdataPropName: !1,
  numberParseOptions: {
    hex: !0,
    leadingZeros: !0,
    eNotation: !0
  },
  tagValueProcessor: function(e, t) {
    return t;
  },
  attributeValueProcessor: function(e, t) {
    return t;
  },
  stopNodes: [],
  //nested tags will not be parsed even for errors
  alwaysCreateTextNode: !1,
  isArray: () => !1,
  commentPropName: !1,
  unpairedTags: [],
  processEntities: !0,
  htmlEntities: !1,
  ignoreDeclaration: !1,
  ignorePiTags: !1,
  transformTagName: !1,
  transformAttributeName: !1,
  updateTag: function(e, t, r) {
    return e;
  }
  // skipEmptyListItem: false
}, ja = function(e) {
  return Object.assign({}, Qr, e);
};
st.buildOptions = ja;
st.defaultOptions = Qr;
class Ia {
  constructor(t) {
    this.tagname = t, this.child = [], this[":@"] = {};
  }
  add(t, r) {
    t === "__proto__" && (t = "#__proto__"), this.child.push({ [t]: r });
  }
  addChild(t) {
    t.tagname === "__proto__" && (t.tagname = "#__proto__"), t[":@"] && Object.keys(t[":@"]).length > 0 ? this.child.push({ [t.tagname]: t.child, [":@"]: t[":@"] }) : this.child.push({ [t.tagname]: t.child });
  }
}
var Ra = Ia;
const Fa = we;
function La(e, t) {
  const r = {};
  if (e[t + 3] === "O" && e[t + 4] === "C" && e[t + 5] === "T" && e[t + 6] === "Y" && e[t + 7] === "P" && e[t + 8] === "E") {
    t = t + 9;
    let n = 1, a = !1, s = !1, l = "";
    for (; t < e.length; t++)
      if (e[t] === "<" && !s) {
        if (a && Ga(e, t))
          t += 7, [entityName, val, t] = Ma(e, t + 1), val.indexOf("&") === -1 && (r[Ka(entityName)] = {
            regx: RegExp(`&${entityName};`, "g"),
            val
          });
        else if (a && Va(e, t))
          t += 8;
        else if (a && Ua(e, t))
          t += 8;
        else if (a && ka(e, t))
          t += 9;
        else if (Ba)
          s = !0;
        else
          throw new Error("Invalid DOCTYPE");
        n++, l = "";
      } else if (e[t] === ">") {
        if (s ? e[t - 1] === "-" && e[t - 2] === "-" && (s = !1, n--) : n--, n === 0)
          break;
      } else
        e[t] === "[" ? a = !0 : l += e[t];
    if (n !== 0)
      throw new Error("Unclosed DOCTYPE");
  } else
    throw new Error("Invalid Tag instead of DOCTYPE");
  return { entities: r, i: t };
}
function Ma(e, t) {
  let r = "";
  for (; t < e.length && e[t] !== "'" && e[t] !== '"'; t++)
    r += e[t];
  if (r = r.trim(), r.indexOf(" ") !== -1)
    throw new Error("External entites are not supported");
  const n = e[t++];
  let a = "";
  for (; t < e.length && e[t] !== n; t++)
    a += e[t];
  return [r, a, t];
}
function Ba(e, t) {
  return e[t + 1] === "!" && e[t + 2] === "-" && e[t + 3] === "-";
}
function Ga(e, t) {
  return e[t + 1] === "!" && e[t + 2] === "E" && e[t + 3] === "N" && e[t + 4] === "T" && e[t + 5] === "I" && e[t + 6] === "T" && e[t + 7] === "Y";
}
function Va(e, t) {
  return e[t + 1] === "!" && e[t + 2] === "E" && e[t + 3] === "L" && e[t + 4] === "E" && e[t + 5] === "M" && e[t + 6] === "E" && e[t + 7] === "N" && e[t + 8] === "T";
}
function Ua(e, t) {
  return e[t + 1] === "!" && e[t + 2] === "A" && e[t + 3] === "T" && e[t + 4] === "T" && e[t + 5] === "L" && e[t + 6] === "I" && e[t + 7] === "S" && e[t + 8] === "T";
}
function ka(e, t) {
  return e[t + 1] === "!" && e[t + 2] === "N" && e[t + 3] === "O" && e[t + 4] === "T" && e[t + 5] === "A" && e[t + 6] === "T" && e[t + 7] === "I" && e[t + 8] === "O" && e[t + 9] === "N";
}
function Ka(e) {
  if (Fa.isName(e))
    return e;
  throw new Error(`Invalid entity name ${e}`);
}
var Xa = La;
const Wa = /^[-+]?0x[a-fA-F0-9]+$/, Ya = /^([\-\+])?(0*)(\.[0-9]+([eE]\-?[0-9]+)?|[0-9]+(\.[0-9]+([eE]\-?[0-9]+)?)?)$/;
!Number.parseInt && window.parseInt && (Number.parseInt = window.parseInt);
!Number.parseFloat && window.parseFloat && (Number.parseFloat = window.parseFloat);
const qa = {
  hex: !0,
  leadingZeros: !0,
  decimalPoint: ".",
  eNotation: !0
  //skipLike: /regex/
};
function Ha(e, t = {}) {
  if (t = Object.assign({}, qa, t), !e || typeof e != "string")
    return e;
  let r = e.trim();
  if (t.skipLike !== void 0 && t.skipLike.test(r))
    return e;
  if (t.hex && Wa.test(r))
    return Number.parseInt(r, 16);
  {
    const n = Ya.exec(r);
    if (n) {
      const a = n[1], s = n[2];
      let l = za(n[3]);
      const u = n[4] || n[6];
      if (!t.leadingZeros && s.length > 0 && a && r[2] !== ".")
        return e;
      if (!t.leadingZeros && s.length > 0 && !a && r[1] !== ".")
        return e;
      {
        const o = Number(r), c = "" + o;
        return c.search(/[eE]/) !== -1 || u ? t.eNotation ? o : e : r.indexOf(".") !== -1 ? c === "0" && l === "" || c === l || a && c === "-" + l ? o : e : s ? l === c || a + l === c ? o : e : r === c || r === a + c ? o : e;
      }
    } else
      return e;
  }
}
function za(e) {
  return e && e.indexOf(".") !== -1 && (e = e.replace(/0+$/, ""), e === "." ? e = "0" : e[0] === "." ? e = "0" + e : e[e.length - 1] === "." && (e = e.substr(0, e.length - 1))), e;
}
var Ja = Ha;
const it = we, le = Ra, Za = Xa, Qa = Ja;
"<((!\\[CDATA\\[([\\s\\S]*?)(]]>))|((NAME:)?(NAME))([^>]*)>|((\\/)(NAME)\\s*>))([^<]*)".replace(/NAME/g, it.nameRegexp);
let Da = class {
  constructor(t) {
    this.options = t, this.currentNode = null, this.tagsNodeStack = [], this.docTypeEntities = {}, this.lastEntities = {
      apos: { regex: /&(apos|#39|#x27);/g, val: "'" },
      gt: { regex: /&(gt|#62|#x3E);/g, val: ">" },
      lt: { regex: /&(lt|#60|#x3C);/g, val: "<" },
      quot: { regex: /&(quot|#34|#x22);/g, val: '"' }
    }, this.ampEntity = { regex: /&(amp|#38|#x26);/g, val: "&" }, this.htmlEntities = {
      space: { regex: /&(nbsp|#160);/g, val: " " },
      // "lt" : { regex: /&(lt|#60);/g, val: "<" },
      // "gt" : { regex: /&(gt|#62);/g, val: ">" },
      // "amp" : { regex: /&(amp|#38);/g, val: "&" },
      // "quot" : { regex: /&(quot|#34);/g, val: "\"" },
      // "apos" : { regex: /&(apos|#39);/g, val: "'" },
      cent: { regex: /&(cent|#162);/g, val: "¢" },
      pound: { regex: /&(pound|#163);/g, val: "£" },
      yen: { regex: /&(yen|#165);/g, val: "¥" },
      euro: { regex: /&(euro|#8364);/g, val: "€" },
      copyright: { regex: /&(copy|#169);/g, val: "©" },
      reg: { regex: /&(reg|#174);/g, val: "®" },
      inr: { regex: /&(inr|#8377);/g, val: "₹" }
    }, this.addExternalEntities = es, this.parseXml = ss, this.parseTextData = ts, this.resolveNameSpace = rs, this.buildAttributesMap = as, this.isItStopNode = ls, this.replaceEntitiesValue = os, this.readStopNodeData = fs, this.saveTextToParentTag = us, this.addChild = is;
  }
};
function es(e) {
  const t = Object.keys(e);
  for (let r = 0; r < t.length; r++) {
    const n = t[r];
    this.lastEntities[n] = {
      regex: new RegExp("&" + n + ";", "g"),
      val: e[n]
    };
  }
}
function ts(e, t, r, n, a, s, l) {
  if (e !== void 0 && (this.options.trimValues && !n && (e = e.trim()), e.length > 0)) {
    l || (e = this.replaceEntitiesValue(e));
    const u = this.options.tagValueProcessor(t, e, r, a, s);
    return u == null ? e : typeof u != typeof e || u !== e ? u : this.options.trimValues ? He(e, this.options.parseTagValue, this.options.numberParseOptions) : e.trim() === e ? He(e, this.options.parseTagValue, this.options.numberParseOptions) : e;
  }
}
function rs(e) {
  if (this.options.removeNSPrefix) {
    const t = e.split(":"), r = e.charAt(0) === "/" ? "/" : "";
    if (t[0] === "xmlns")
      return "";
    t.length === 2 && (e = r + t[1]);
  }
  return e;
}
const ns = new RegExp(`([^\\s=]+)\\s*(=\\s*(['"])([\\s\\S]*?)\\3)?`, "gm");
function as(e, t, r) {
  if (!this.options.ignoreAttributes && typeof e == "string") {
    const n = it.getAllMatches(e, ns), a = n.length, s = {};
    for (let l = 0; l < a; l++) {
      const u = this.resolveNameSpace(n[l][1]);
      let o = n[l][4], c = this.options.attributeNamePrefix + u;
      if (u.length)
        if (this.options.transformAttributeName && (c = this.options.transformAttributeName(c)), c === "__proto__" && (c = "#__proto__"), o !== void 0) {
          this.options.trimValues && (o = o.trim()), o = this.replaceEntitiesValue(o);
          const d = this.options.attributeValueProcessor(u, o, t);
          d == null ? s[c] = o : typeof d != typeof o || d !== o ? s[c] = d : s[c] = He(
            o,
            this.options.parseAttributeValue,
            this.options.numberParseOptions
          );
        } else
          this.options.allowBooleanAttributes && (s[c] = !0);
    }
    if (!Object.keys(s).length)
      return;
    if (this.options.attributesGroupName) {
      const l = {};
      return l[this.options.attributesGroupName] = s, l;
    }
    return s;
  }
}
const ss = function(e) {
  e = e.replace(/\r\n?/g, `
`);
  const t = new le("!xml");
  let r = t, n = "", a = "";
  for (let s = 0; s < e.length; s++)
    if (e[s] === "<")
      if (e[s + 1] === "/") {
        const u = W(e, ">", s, "Closing Tag is not closed.");
        let o = e.substring(s + 2, u).trim();
        if (this.options.removeNSPrefix) {
          const g = o.indexOf(":");
          g !== -1 && (o = o.substr(g + 1));
        }
        this.options.transformTagName && (o = this.options.transformTagName(o)), r && (n = this.saveTextToParentTag(n, r, a));
        const c = a.substring(a.lastIndexOf(".") + 1);
        if (o && this.options.unpairedTags.indexOf(o) !== -1)
          throw new Error(`Unpaired tag can not be used as closing tag: </${o}>`);
        let d = 0;
        c && this.options.unpairedTags.indexOf(c) !== -1 ? (d = a.lastIndexOf(".", a.lastIndexOf(".") - 1), this.tagsNodeStack.pop()) : d = a.lastIndexOf("."), a = a.substring(0, d), r = this.tagsNodeStack.pop(), n = "", s = u;
      } else if (e[s + 1] === "?") {
        let u = qe(e, s, !1, "?>");
        if (!u)
          throw new Error("Pi Tag is not closed.");
        if (n = this.saveTextToParentTag(n, r, a), !(this.options.ignoreDeclaration && u.tagName === "?xml" || this.options.ignorePiTags)) {
          const o = new le(u.tagName);
          o.add(this.options.textNodeName, ""), u.tagName !== u.tagExp && u.attrExpPresent && (o[":@"] = this.buildAttributesMap(u.tagExp, a, u.tagName)), this.addChild(r, o, a);
        }
        s = u.closeIndex + 1;
      } else if (e.substr(s + 1, 3) === "!--") {
        const u = W(e, "-->", s + 4, "Comment is not closed.");
        if (this.options.commentPropName) {
          const o = e.substring(s + 4, u - 2);
          n = this.saveTextToParentTag(n, r, a), r.add(this.options.commentPropName, [{ [this.options.textNodeName]: o }]);
        }
        s = u;
      } else if (e.substr(s + 1, 2) === "!D") {
        const u = Za(e, s);
        this.docTypeEntities = u.entities, s = u.i;
      } else if (e.substr(s + 1, 2) === "![") {
        const u = W(e, "]]>", s, "CDATA is not closed.") - 2, o = e.substring(s + 9, u);
        if (n = this.saveTextToParentTag(n, r, a), this.options.cdataPropName)
          r.add(this.options.cdataPropName, [{ [this.options.textNodeName]: o }]);
        else {
          let c = this.parseTextData(o, r.tagname, a, !0, !1, !0);
          c == null && (c = ""), r.add(this.options.textNodeName, c);
        }
        s = u + 2;
      } else {
        let u = qe(e, s, this.options.removeNSPrefix), o = u.tagName, c = u.tagExp, d = u.attrExpPresent, g = u.closeIndex;
        this.options.transformTagName && (o = this.options.transformTagName(o)), r && n && r.tagname !== "!xml" && (n = this.saveTextToParentTag(n, r, a, !1));
        const h = r;
        if (h && this.options.unpairedTags.indexOf(h.tagname) !== -1 && (r = this.tagsNodeStack.pop(), a = a.substring(0, a.lastIndexOf("."))), o !== t.tagname && (a += a ? "." + o : o), this.isItStopNode(this.options.stopNodes, a, o)) {
          let v = "";
          if (c.length > 0 && c.lastIndexOf("/") === c.length - 1)
            s = u.closeIndex;
          else if (this.options.unpairedTags.indexOf(o) !== -1)
            s = u.closeIndex;
          else {
            const w = this.readStopNodeData(e, o, g + 1);
            if (!w)
              throw new Error(`Unexpected end of ${o}`);
            s = w.i, v = w.tagContent;
          }
          const T = new le(o);
          o !== c && d && (T[":@"] = this.buildAttributesMap(c, a, o)), v && (v = this.parseTextData(v, o, a, !0, d, !0, !0)), a = a.substr(0, a.lastIndexOf(".")), T.add(this.options.textNodeName, v), this.addChild(r, T, a);
        } else {
          if (c.length > 0 && c.lastIndexOf("/") === c.length - 1) {
            o[o.length - 1] === "/" ? (o = o.substr(0, o.length - 1), c = o) : c = c.substr(0, c.length - 1), this.options.transformTagName && (o = this.options.transformTagName(o));
            const v = new le(o);
            o !== c && d && (v[":@"] = this.buildAttributesMap(c, a, o)), this.addChild(r, v, a), a = a.substr(0, a.lastIndexOf("."));
          } else {
            const v = new le(o);
            this.tagsNodeStack.push(r), o !== c && d && (v[":@"] = this.buildAttributesMap(c, a, o)), this.addChild(r, v, a), r = v;
          }
          n = "", s = g;
        }
      }
    else
      n += e[s];
  return t.child;
};
function is(e, t, r) {
  const n = this.options.updateTag(t.tagname, r, t[":@"]);
  n === !1 || (typeof n == "string" && (t.tagname = n), e.addChild(t));
}
const os = function(e) {
  if (this.options.processEntities) {
    for (let t in this.docTypeEntities) {
      const r = this.docTypeEntities[t];
      e = e.replace(r.regx, r.val);
    }
    for (let t in this.lastEntities) {
      const r = this.lastEntities[t];
      e = e.replace(r.regex, r.val);
    }
    if (this.options.htmlEntities)
      for (let t in this.htmlEntities) {
        const r = this.htmlEntities[t];
        e = e.replace(r.regex, r.val);
      }
    e = e.replace(this.ampEntity.regex, this.ampEntity.val);
  }
  return e;
};
function us(e, t, r, n) {
  return e && (n === void 0 && (n = Object.keys(t.child).length === 0), e = this.parseTextData(
    e,
    t.tagname,
    r,
    !1,
    t[":@"] ? Object.keys(t[":@"]).length !== 0 : !1,
    n
  ), e !== void 0 && e !== "" && t.add(this.options.textNodeName, e), e = ""), e;
}
function ls(e, t, r) {
  const n = "*." + r;
  for (const a in e) {
    const s = e[a];
    if (n === s || t === s)
      return !0;
  }
  return !1;
}
function cs(e, t, r = ">") {
  let n, a = "";
  for (let s = t; s < e.length; s++) {
    let l = e[s];
    if (n)
      l === n && (n = "");
    else if (l === '"' || l === "'")
      n = l;
    else if (l === r[0])
      if (r[1]) {
        if (e[s + 1] === r[1])
          return {
            data: a,
            index: s
          };
      } else
        return {
          data: a,
          index: s
        };
    else
      l === "	" && (l = " ");
    a += l;
  }
}
function W(e, t, r, n) {
  const a = e.indexOf(t, r);
  if (a === -1)
    throw new Error(n);
  return a + t.length - 1;
}
function qe(e, t, r, n = ">") {
  const a = cs(e, t + 1, n);
  if (!a)
    return;
  let s = a.data;
  const l = a.index, u = s.search(/\s/);
  let o = s, c = !0;
  if (u !== -1 && (o = s.substr(0, u).replace(/\s\s*$/, ""), s = s.substr(u + 1)), r) {
    const d = o.indexOf(":");
    d !== -1 && (o = o.substr(d + 1), c = o !== a.data.substr(d + 1));
  }
  return {
    tagName: o,
    tagExp: s,
    closeIndex: l,
    attrExpPresent: c
  };
}
function fs(e, t, r) {
  const n = r;
  let a = 1;
  for (; r < e.length; r++)
    if (e[r] === "<")
      if (e[r + 1] === "/") {
        const s = W(e, ">", r, `${t} is not closed`);
        if (e.substring(r + 2, s).trim() === t && (a--, a === 0))
          return {
            tagContent: e.substring(n, r),
            i: s
          };
        r = s;
      } else if (e[r + 1] === "?")
        r = W(e, "?>", r + 1, "StopNode is not closed.");
      else if (e.substr(r + 1, 3) === "!--")
        r = W(e, "-->", r + 3, "StopNode is not closed.");
      else if (e.substr(r + 1, 2) === "![")
        r = W(e, "]]>", r, "StopNode is not closed.") - 2;
      else {
        const s = qe(e, r, ">");
        s && ((s && s.tagName) === t && s.tagExp[s.tagExp.length - 1] !== "/" && a++, r = s.closeIndex);
      }
}
function He(e, t, r) {
  if (t && typeof e == "string") {
    const n = e.trim();
    return n === "true" ? !0 : n === "false" ? !1 : Qa(e, r);
  } else
    return it.isExist(e) ? e : "";
}
var ds = Da, Dr = {};
function ps(e, t) {
  return en(e, t);
}
function en(e, t, r) {
  let n;
  const a = {};
  for (let s = 0; s < e.length; s++) {
    const l = e[s], u = gs(l);
    let o = "";
    if (r === void 0 ? o = u : o = r + "." + u, u === t.textNodeName)
      n === void 0 ? n = l[u] : n += "" + l[u];
    else {
      if (u === void 0)
        continue;
      if (l[u]) {
        let c = en(l[u], t, o);
        const d = vs(c, t);
        l[":@"] ? hs(c, l[":@"], o, t) : Object.keys(c).length === 1 && c[t.textNodeName] !== void 0 && !t.alwaysCreateTextNode ? c = c[t.textNodeName] : Object.keys(c).length === 0 && (t.alwaysCreateTextNode ? c[t.textNodeName] = "" : c = ""), a[u] !== void 0 && a.hasOwnProperty(u) ? (Array.isArray(a[u]) || (a[u] = [a[u]]), a[u].push(c)) : t.isArray(u, o, d) ? a[u] = [c] : a[u] = c;
      }
    }
  }
  return typeof n == "string" ? n.length > 0 && (a[t.textNodeName] = n) : n !== void 0 && (a[t.textNodeName] = n), a;
}
function gs(e) {
  const t = Object.keys(e);
  for (let r = 0; r < t.length; r++) {
    const n = t[r];
    if (n !== ":@")
      return n;
  }
}
function hs(e, t, r, n) {
  if (t) {
    const a = Object.keys(t), s = a.length;
    for (let l = 0; l < s; l++) {
      const u = a[l];
      n.isArray(u, r + "." + u, !0, !0) ? e[u] = [t[u]] : e[u] = t[u];
    }
  }
}
function vs(e, t) {
  const { textNodeName: r } = t, n = Object.keys(e).length;
  return !!(n === 0 || n === 1 && (e[r] || typeof e[r] == "boolean" || e[r] === 0));
}
Dr.prettify = ps;
const { buildOptions: bs } = st, ys = ds, { prettify: $s } = Dr, _s = nt;
let Ts = class {
  constructor(t) {
    this.externalEntities = {}, this.options = bs(t);
  }
  /**
   * Parse XML dats to JS object 
   * @param {string|Buffer} xmlData 
   * @param {boolean|Object} validationOption 
   */
  parse(t, r) {
    if (typeof t != "string")
      if (t.toString)
        t = t.toString();
      else
        throw new Error("XML data is accepted in String or Bytes[] form.");
    if (r) {
      r === !0 && (r = {});
      const s = _s.validate(t, r);
      if (s !== !0)
        throw Error(`${s.err.msg}:${s.err.line}:${s.err.col}`);
    }
    const n = new ys(this.options);
    n.addExternalEntities(this.externalEntities);
    const a = n.parseXml(t);
    return this.options.preserveOrder || a === void 0 ? a : $s(a, this.options);
  }
  /**
   * Add Entity which is not by default supported by this library
   * @param {string} key 
   * @param {string} value 
   */
  addEntity(t, r) {
    if (r.indexOf("&") !== -1)
      throw new Error("Entity value can't have '&'");
    if (t.indexOf("&") !== -1 || t.indexOf(";") !== -1)
      throw new Error("An entity must be set without '&' and ';'. Eg. use '#xD' for '&#xD;'");
    if (r === "&")
      throw new Error("An entity with value '&' is not permitted");
    this.externalEntities[t] = r;
  }
};
var ms = Ts;
const Es = `
`;
function As(e, t) {
  let r = "";
  return t.format && t.indentBy.length > 0 && (r = Es), tn(e, t, "", r);
}
function tn(e, t, r, n) {
  let a = "", s = !1;
  for (let l = 0; l < e.length; l++) {
    const u = e[l], o = Os(u);
    let c = "";
    if (r.length === 0 ? c = o : c = `${r}.${o}`, o === t.textNodeName) {
      let T = u[o];
      Cs(c, t) || (T = t.tagValueProcessor(o, T), T = rn(T, t)), s && (a += n), a += T, s = !1;
      continue;
    } else if (o === t.cdataPropName) {
      s && (a += n), a += `<![CDATA[${u[o][0][t.textNodeName]}]]>`, s = !1;
      continue;
    } else if (o === t.commentPropName) {
      a += n + `<!--${u[o][0][t.textNodeName]}-->`, s = !0;
      continue;
    } else if (o[0] === "?") {
      const T = tr(u[":@"], t), w = o === "?xml" ? "" : n;
      let C = u[o][0][t.textNodeName];
      C = C.length !== 0 ? " " + C : "", a += w + `<${o}${C}${T}?>`, s = !0;
      continue;
    }
    let d = n;
    d !== "" && (d += t.indentBy);
    const g = tr(u[":@"], t), h = n + `<${o}${g}`, v = tn(u[o], t, c, d);
    t.unpairedTags.indexOf(o) !== -1 ? t.suppressUnpairedNode ? a += h + ">" : a += h + "/>" : (!v || v.length === 0) && t.suppressEmptyNode ? a += h + "/>" : v && v.endsWith(">") ? a += h + `>${v}${n}</${o}>` : (a += h + ">", v && n !== "" && (v.includes("/>") || v.includes("</")) ? a += n + t.indentBy + v + n : a += v, a += `</${o}>`), s = !0;
  }
  return a;
}
function Os(e) {
  const t = Object.keys(e);
  for (let r = 0; r < t.length; r++) {
    const n = t[r];
    if (n !== ":@")
      return n;
  }
}
function tr(e, t) {
  let r = "";
  if (e && !t.ignoreAttributes)
    for (let n in e) {
      let a = t.attributeValueProcessor(n, e[n]);
      a = rn(a, t), a === !0 && t.suppressBooleanAttributes ? r += ` ${n.substr(t.attributeNamePrefix.length)}` : r += ` ${n.substr(t.attributeNamePrefix.length)}="${a}"`;
    }
  return r;
}
function Cs(e, t) {
  e = e.substr(0, e.length - t.textNodeName.length - 1);
  let r = e.substr(e.lastIndexOf(".") + 1);
  for (let n in t.stopNodes)
    if (t.stopNodes[n] === e || t.stopNodes[n] === "*." + r)
      return !0;
  return !1;
}
function rn(e, t) {
  if (e && e.length > 0 && t.processEntities)
    for (let r = 0; r < t.entities.length; r++) {
      const n = t.entities[r];
      e = e.replace(n.regex, n.val);
    }
  return e;
}
var Ns = As;
const ws = Ns, Ps = {
  attributeNamePrefix: "@_",
  attributesGroupName: !1,
  textNodeName: "#text",
  ignoreAttributes: !0,
  cdataPropName: !1,
  format: !1,
  indentBy: "  ",
  suppressEmptyNode: !1,
  suppressUnpairedNode: !0,
  suppressBooleanAttributes: !0,
  tagValueProcessor: function(e, t) {
    return t;
  },
  attributeValueProcessor: function(e, t) {
    return t;
  },
  preserveOrder: !1,
  commentPropName: !1,
  unpairedTags: [],
  entities: [
    { regex: new RegExp("&", "g"), val: "&amp;" },
    //it must be on top
    { regex: new RegExp(">", "g"), val: "&gt;" },
    { regex: new RegExp("<", "g"), val: "&lt;" },
    { regex: new RegExp("'", "g"), val: "&apos;" },
    { regex: new RegExp('"', "g"), val: "&quot;" }
  ],
  processEntities: !0,
  stopNodes: [],
  // transformTagName: false,
  // transformAttributeName: false,
  oneListGroup: !1
};
function V(e) {
  this.options = Object.assign({}, Ps, e), this.options.ignoreAttributes || this.options.attributesGroupName ? this.isAttribute = function() {
    return !1;
  } : (this.attrPrefixLen = this.options.attributeNamePrefix.length, this.isAttribute = js), this.processTextOrObjNode = Ss, this.options.format ? (this.indentate = xs, this.tagEndChar = `>
`, this.newLine = `
`) : (this.indentate = function() {
    return "";
  }, this.tagEndChar = ">", this.newLine = "");
}
V.prototype.build = function(e) {
  return this.options.preserveOrder ? ws(e, this.options) : (Array.isArray(e) && this.options.arrayNodeName && this.options.arrayNodeName.length > 1 && (e = {
    [this.options.arrayNodeName]: e
  }), this.j2x(e, 0).val);
};
V.prototype.j2x = function(e, t) {
  let r = "", n = "";
  for (let a in e)
    if (!(typeof e[a] > "u"))
      if (e[a] === null)
        a[0] === "?" ? n += this.indentate(t) + "<" + a + "?" + this.tagEndChar : n += this.indentate(t) + "<" + a + "/" + this.tagEndChar;
      else if (e[a] instanceof Date)
        n += this.buildTextValNode(e[a], a, "", t);
      else if (typeof e[a] != "object") {
        const s = this.isAttribute(a);
        if (s)
          r += this.buildAttrPairStr(s, "" + e[a]);
        else if (a === this.options.textNodeName) {
          let l = this.options.tagValueProcessor(a, "" + e[a]);
          n += this.replaceEntitiesValue(l);
        } else
          n += this.buildTextValNode(e[a], a, "", t);
      } else if (Array.isArray(e[a])) {
        const s = e[a].length;
        let l = "";
        for (let u = 0; u < s; u++) {
          const o = e[a][u];
          typeof o > "u" || (o === null ? a[0] === "?" ? n += this.indentate(t) + "<" + a + "?" + this.tagEndChar : n += this.indentate(t) + "<" + a + "/" + this.tagEndChar : typeof o == "object" ? this.options.oneListGroup ? l += this.j2x(o, t + 1).val : l += this.processTextOrObjNode(o, a, t) : l += this.buildTextValNode(o, a, "", t));
        }
        this.options.oneListGroup && (l = this.buildObjectNode(l, a, "", t)), n += l;
      } else if (this.options.attributesGroupName && a === this.options.attributesGroupName) {
        const s = Object.keys(e[a]), l = s.length;
        for (let u = 0; u < l; u++)
          r += this.buildAttrPairStr(s[u], "" + e[a][s[u]]);
      } else
        n += this.processTextOrObjNode(e[a], a, t);
  return { attrStr: r, val: n };
};
V.prototype.buildAttrPairStr = function(e, t) {
  return t = this.options.attributeValueProcessor(e, "" + t), t = this.replaceEntitiesValue(t), this.options.suppressBooleanAttributes && t === "true" ? " " + e : " " + e + '="' + t + '"';
};
function Ss(e, t, r) {
  const n = this.j2x(e, r + 1);
  return e[this.options.textNodeName] !== void 0 && Object.keys(e).length === 1 ? this.buildTextValNode(e[this.options.textNodeName], t, n.attrStr, r) : this.buildObjectNode(n.val, t, n.attrStr, r);
}
V.prototype.buildObjectNode = function(e, t, r, n) {
  if (e === "")
    return t[0] === "?" ? this.indentate(n) + "<" + t + r + "?" + this.tagEndChar : this.indentate(n) + "<" + t + r + this.closeTag(t) + this.tagEndChar;
  {
    let a = "</" + t + this.tagEndChar, s = "";
    return t[0] === "?" && (s = "?", a = ""), r && e.indexOf("<") === -1 ? this.indentate(n) + "<" + t + r + s + ">" + e + a : this.options.commentPropName !== !1 && t === this.options.commentPropName && s.length === 0 ? this.indentate(n) + `<!--${e}-->` + this.newLine : this.indentate(n) + "<" + t + r + s + this.tagEndChar + e + this.indentate(n) + a;
  }
};
V.prototype.closeTag = function(e) {
  let t = "";
  return this.options.unpairedTags.indexOf(e) !== -1 ? this.options.suppressUnpairedNode || (t = "/") : this.options.suppressEmptyNode ? t = "/" : t = `></${e}`, t;
};
V.prototype.buildTextValNode = function(e, t, r, n) {
  if (this.options.cdataPropName !== !1 && t === this.options.cdataPropName)
    return this.indentate(n) + `<![CDATA[${e}]]>` + this.newLine;
  if (this.options.commentPropName !== !1 && t === this.options.commentPropName)
    return this.indentate(n) + `<!--${e}-->` + this.newLine;
  if (t[0] === "?")
    return this.indentate(n) + "<" + t + r + "?" + this.tagEndChar;
  {
    let a = this.options.tagValueProcessor(t, e);
    return a = this.replaceEntitiesValue(a), a === "" ? this.indentate(n) + "<" + t + r + this.closeTag(t) + this.tagEndChar : this.indentate(n) + "<" + t + r + ">" + a + "</" + t + this.tagEndChar;
  }
};
V.prototype.replaceEntitiesValue = function(e) {
  if (e && e.length > 0 && this.options.processEntities)
    for (let t = 0; t < this.options.entities.length; t++) {
      const r = this.options.entities[t];
      e = e.replace(r.regex, r.val);
    }
  return e;
};
function xs(e) {
  return this.options.indentBy.repeat(e);
}
function js(e) {
  return e.startsWith(this.options.attributeNamePrefix) ? e.substr(this.attrPrefixLen) : !1;
}
var Is = V;
const Rs = nt, Fs = ms, Ls = Is;
var Ms = {
  XMLParser: Fs,
  XMLValidator: Rs,
  XMLBuilder: Ls
};
const Bs = new Ms.XMLParser({
  preserveOrder: !0,
  ignoreAttributes: !1,
  attributeNamePrefix: "",
  allowBooleanAttributes: !0,
  commentPropName: tt,
  cdataPropName: rt,
  parseTagValue: !1
});
function Gs(e) {
  return qr(() => {
    try {
      return { json: Bs.parse(e), valid: !0 };
    } catch (t) {
      return { json: null, valid: !1, errorMessage: `Fail to parse: ${t.message}` };
    }
  }, [e]);
}
function Vs() {
  this.__data__ = [], this.size = 0;
}
var Us = Vs;
function ks(e, t) {
  return e === t || e !== e && t !== t;
}
var ot = ks, Ks = ot;
function Xs(e, t) {
  for (var r = e.length; r--; )
    if (Ks(e[r][0], t))
      return r;
  return -1;
}
var Pe = Xs, Ws = Pe, Ys = Array.prototype, qs = Ys.splice;
function Hs(e) {
  var t = this.__data__, r = Ws(t, e);
  if (r < 0)
    return !1;
  var n = t.length - 1;
  return r == n ? t.pop() : qs.call(t, r, 1), --this.size, !0;
}
var zs = Hs, Js = Pe;
function Zs(e) {
  var t = this.__data__, r = Js(t, e);
  return r < 0 ? void 0 : t[r][1];
}
var Qs = Zs, Ds = Pe;
function ei(e) {
  return Ds(this.__data__, e) > -1;
}
var ti = ei, ri = Pe;
function ni(e, t) {
  var r = this.__data__, n = ri(r, e);
  return n < 0 ? (++this.size, r.push([e, t])) : r[n][1] = t, this;
}
var ai = ni, si = Us, ii = zs, oi = Qs, ui = ti, li = ai;
function Z(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
Z.prototype.clear = si;
Z.prototype.delete = ii;
Z.prototype.get = oi;
Z.prototype.has = ui;
Z.prototype.set = li;
var Se = Z, ci = Se;
function fi() {
  this.__data__ = new ci(), this.size = 0;
}
var di = fi;
function pi(e) {
  var t = this.__data__, r = t.delete(e);
  return this.size = t.size, r;
}
var gi = pi;
function hi(e) {
  return this.__data__.get(e);
}
var vi = hi;
function bi(e) {
  return this.__data__.has(e);
}
var yi = bi, $i = typeof $e == "object" && $e && $e.Object === Object && $e, nn = $i, _i = nn, Ti = typeof self == "object" && self && self.Object === Object && self, mi = _i || Ti || Function("return this")(), L = mi, Ei = L, Ai = Ei.Symbol, Q = Ai, rr = Q, an = Object.prototype, Oi = an.hasOwnProperty, Ci = an.toString, ce = rr ? rr.toStringTag : void 0;
function Ni(e) {
  var t = Oi.call(e, ce), r = e[ce];
  try {
    e[ce] = void 0;
    var n = !0;
  } catch {
  }
  var a = Ci.call(e);
  return n && (t ? e[ce] = r : delete e[ce]), a;
}
var wi = Ni, Pi = Object.prototype, Si = Pi.toString;
function xi(e) {
  return Si.call(e);
}
var ji = xi, nr = Q, Ii = wi, Ri = ji, Fi = "[object Null]", Li = "[object Undefined]", ar = nr ? nr.toStringTag : void 0;
function Mi(e) {
  return e == null ? e === void 0 ? Li : Fi : ar && ar in Object(e) ? Ii(e) : Ri(e);
}
var D = Mi;
function Bi(e) {
  var t = typeof e;
  return e != null && (t == "object" || t == "function");
}
var fe = Bi, Gi = D, Vi = fe, Ui = "[object AsyncFunction]", ki = "[object Function]", Ki = "[object GeneratorFunction]", Xi = "[object Proxy]";
function Wi(e) {
  if (!Vi(e))
    return !1;
  var t = Gi(e);
  return t == ki || t == Ki || t == Ui || t == Xi;
}
var sn = Wi, Yi = L, qi = Yi["__core-js_shared__"], Hi = qi, ke = Hi, sr = function() {
  var e = /[^.]+$/.exec(ke && ke.keys && ke.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function zi(e) {
  return !!sr && sr in e;
}
var Ji = zi, Zi = Function.prototype, Qi = Zi.toString;
function Di(e) {
  if (e != null) {
    try {
      return Qi.call(e);
    } catch {
    }
    try {
      return e + "";
    } catch {
    }
  }
  return "";
}
var on = Di, eo = sn, to = Ji, ro = fe, no = on, ao = /[\\^$.*+?()[\]{}|]/g, so = /^\[object .+?Constructor\]$/, io = Function.prototype, oo = Object.prototype, uo = io.toString, lo = oo.hasOwnProperty, co = RegExp(
  "^" + uo.call(lo).replace(ao, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function fo(e) {
  if (!ro(e) || to(e))
    return !1;
  var t = eo(e) ? co : so;
  return t.test(no(e));
}
var po = fo;
function go(e, t) {
  return e == null ? void 0 : e[t];
}
var ho = go, vo = po, bo = ho;
function yo(e, t) {
  var r = bo(e, t);
  return vo(r) ? r : void 0;
}
var Y = yo, $o = Y, _o = L, To = $o(_o, "Map"), ut = To, mo = Y, Eo = mo(Object, "create"), xe = Eo, ir = xe;
function Ao() {
  this.__data__ = ir ? ir(null) : {}, this.size = 0;
}
var Oo = Ao;
function Co(e) {
  var t = this.has(e) && delete this.__data__[e];
  return this.size -= t ? 1 : 0, t;
}
var No = Co, wo = xe, Po = "__lodash_hash_undefined__", So = Object.prototype, xo = So.hasOwnProperty;
function jo(e) {
  var t = this.__data__;
  if (wo) {
    var r = t[e];
    return r === Po ? void 0 : r;
  }
  return xo.call(t, e) ? t[e] : void 0;
}
var Io = jo, Ro = xe, Fo = Object.prototype, Lo = Fo.hasOwnProperty;
function Mo(e) {
  var t = this.__data__;
  return Ro ? t[e] !== void 0 : Lo.call(t, e);
}
var Bo = Mo, Go = xe, Vo = "__lodash_hash_undefined__";
function Uo(e, t) {
  var r = this.__data__;
  return this.size += this.has(e) ? 0 : 1, r[e] = Go && t === void 0 ? Vo : t, this;
}
var ko = Uo, Ko = Oo, Xo = No, Wo = Io, Yo = Bo, qo = ko;
function ee(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
ee.prototype.clear = Ko;
ee.prototype.delete = Xo;
ee.prototype.get = Wo;
ee.prototype.has = Yo;
ee.prototype.set = qo;
var Ho = ee, or = Ho, zo = Se, Jo = ut;
function Zo() {
  this.size = 0, this.__data__ = {
    hash: new or(),
    map: new (Jo || zo)(),
    string: new or()
  };
}
var Qo = Zo;
function Do(e) {
  var t = typeof e;
  return t == "string" || t == "number" || t == "symbol" || t == "boolean" ? e !== "__proto__" : e === null;
}
var eu = Do, tu = eu;
function ru(e, t) {
  var r = e.__data__;
  return tu(t) ? r[typeof t == "string" ? "string" : "hash"] : r.map;
}
var je = ru, nu = je;
function au(e) {
  var t = nu(this, e).delete(e);
  return this.size -= t ? 1 : 0, t;
}
var su = au, iu = je;
function ou(e) {
  return iu(this, e).get(e);
}
var uu = ou, lu = je;
function cu(e) {
  return lu(this, e).has(e);
}
var fu = cu, du = je;
function pu(e, t) {
  var r = du(this, e), n = r.size;
  return r.set(e, t), this.size += r.size == n ? 0 : 1, this;
}
var gu = pu, hu = Qo, vu = su, bu = uu, yu = fu, $u = gu;
function te(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
te.prototype.clear = hu;
te.prototype.delete = vu;
te.prototype.get = bu;
te.prototype.has = yu;
te.prototype.set = $u;
var lt = te, _u = Se, Tu = ut, mu = lt, Eu = 200;
function Au(e, t) {
  var r = this.__data__;
  if (r instanceof _u) {
    var n = r.__data__;
    if (!Tu || n.length < Eu - 1)
      return n.push([e, t]), this.size = ++r.size, this;
    r = this.__data__ = new mu(n);
  }
  return r.set(e, t), this.size = r.size, this;
}
var Ou = Au, Cu = Se, Nu = di, wu = gi, Pu = vi, Su = yi, xu = Ou;
function re(e) {
  var t = this.__data__ = new Cu(e);
  this.size = t.size;
}
re.prototype.clear = Nu;
re.prototype.delete = wu;
re.prototype.get = Pu;
re.prototype.has = Su;
re.prototype.set = xu;
var un = re, ju = "__lodash_hash_undefined__";
function Iu(e) {
  return this.__data__.set(e, ju), this;
}
var Ru = Iu;
function Fu(e) {
  return this.__data__.has(e);
}
var Lu = Fu, Mu = lt, Bu = Ru, Gu = Lu;
function Ae(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.__data__ = new Mu(); ++t < r; )
    this.add(e[t]);
}
Ae.prototype.add = Ae.prototype.push = Bu;
Ae.prototype.has = Gu;
var Vu = Ae;
function Uu(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length; ++r < n; )
    if (t(e[r], r, e))
      return !0;
  return !1;
}
var ku = Uu;
function Ku(e, t) {
  return e.has(t);
}
var Xu = Ku, Wu = Vu, Yu = ku, qu = Xu, Hu = 1, zu = 2;
function Ju(e, t, r, n, a, s) {
  var l = r & Hu, u = e.length, o = t.length;
  if (u != o && !(l && o > u))
    return !1;
  var c = s.get(e), d = s.get(t);
  if (c && d)
    return c == t && d == e;
  var g = -1, h = !0, v = r & zu ? new Wu() : void 0;
  for (s.set(e, t), s.set(t, e); ++g < u; ) {
    var T = e[g], w = t[g];
    if (n)
      var C = l ? n(w, T, g, t, e, s) : n(T, w, g, e, t, s);
    if (C !== void 0) {
      if (C)
        continue;
      h = !1;
      break;
    }
    if (v) {
      if (!Yu(t, function(P, N) {
        if (!qu(v, N) && (T === P || a(T, P, r, n, s)))
          return v.push(N);
      })) {
        h = !1;
        break;
      }
    } else if (!(T === w || a(T, w, r, n, s))) {
      h = !1;
      break;
    }
  }
  return s.delete(e), s.delete(t), h;
}
var ln = Ju, Zu = L, Qu = Zu.Uint8Array, cn = Qu;
function Du(e) {
  var t = -1, r = Array(e.size);
  return e.forEach(function(n, a) {
    r[++t] = [a, n];
  }), r;
}
var el = Du;
function tl(e) {
  var t = -1, r = Array(e.size);
  return e.forEach(function(n) {
    r[++t] = n;
  }), r;
}
var rl = tl, ur = Q, lr = cn, nl = ot, al = ln, sl = el, il = rl, ol = 1, ul = 2, ll = "[object Boolean]", cl = "[object Date]", fl = "[object Error]", dl = "[object Map]", pl = "[object Number]", gl = "[object RegExp]", hl = "[object Set]", vl = "[object String]", bl = "[object Symbol]", yl = "[object ArrayBuffer]", $l = "[object DataView]", cr = ur ? ur.prototype : void 0, Ke = cr ? cr.valueOf : void 0;
function _l(e, t, r, n, a, s, l) {
  switch (r) {
    case $l:
      if (e.byteLength != t.byteLength || e.byteOffset != t.byteOffset)
        return !1;
      e = e.buffer, t = t.buffer;
    case yl:
      return !(e.byteLength != t.byteLength || !s(new lr(e), new lr(t)));
    case ll:
    case cl:
    case pl:
      return nl(+e, +t);
    case fl:
      return e.name == t.name && e.message == t.message;
    case gl:
    case vl:
      return e == t + "";
    case dl:
      var u = sl;
    case hl:
      var o = n & ol;
      if (u || (u = il), e.size != t.size && !o)
        return !1;
      var c = l.get(e);
      if (c)
        return c == t;
      n |= ul, l.set(e, t);
      var d = al(u(e), u(t), n, a, s, l);
      return l.delete(e), d;
    case bl:
      if (Ke)
        return Ke.call(e) == Ke.call(t);
  }
  return !1;
}
var Tl = _l;
function ml(e, t) {
  for (var r = -1, n = t.length, a = e.length; ++r < n; )
    e[a + r] = t[r];
  return e;
}
var ct = ml, El = Array.isArray, U = El, Al = ct, Ol = U;
function Cl(e, t, r) {
  var n = t(e);
  return Ol(e) ? n : Al(n, r(e));
}
var fn = Cl;
function Nl(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length, a = 0, s = []; ++r < n; ) {
    var l = e[r];
    t(l, r, e) && (s[a++] = l);
  }
  return s;
}
var wl = Nl;
function Pl() {
  return [];
}
var dn = Pl, Sl = wl, xl = dn, jl = Object.prototype, Il = jl.propertyIsEnumerable, fr = Object.getOwnPropertySymbols, Rl = fr ? function(e) {
  return e == null ? [] : (e = Object(e), Sl(fr(e), function(t) {
    return Il.call(e, t);
  }));
} : xl, ft = Rl;
function Fl(e, t) {
  for (var r = -1, n = Array(e); ++r < e; )
    n[r] = t(r);
  return n;
}
var Ll = Fl;
function Ml(e) {
  return e != null && typeof e == "object";
}
var k = Ml, Bl = D, Gl = k, Vl = "[object Arguments]";
function Ul(e) {
  return Gl(e) && Bl(e) == Vl;
}
var kl = Ul, dr = kl, Kl = k, pn = Object.prototype, Xl = pn.hasOwnProperty, Wl = pn.propertyIsEnumerable, Yl = dr(function() {
  return arguments;
}()) ? dr : function(e) {
  return Kl(e) && Xl.call(e, "callee") && !Wl.call(e, "callee");
}, gn = Yl, Oe = { exports: {} };
function ql() {
  return !1;
}
var Hl = ql;
Oe.exports;
(function(e, t) {
  var r = L, n = Hl, a = t && !t.nodeType && t, s = a && !0 && e && !e.nodeType && e, l = s && s.exports === a, u = l ? r.Buffer : void 0, o = u ? u.isBuffer : void 0, c = o || n;
  e.exports = c;
})(Oe, Oe.exports);
var dt = Oe.exports, zl = 9007199254740991, Jl = /^(?:0|[1-9]\d*)$/;
function Zl(e, t) {
  var r = typeof e;
  return t = t ?? zl, !!t && (r == "number" || r != "symbol" && Jl.test(e)) && e > -1 && e % 1 == 0 && e < t;
}
var Ql = Zl, Dl = 9007199254740991;
function ec(e) {
  return typeof e == "number" && e > -1 && e % 1 == 0 && e <= Dl;
}
var hn = ec, tc = D, rc = hn, nc = k, ac = "[object Arguments]", sc = "[object Array]", ic = "[object Boolean]", oc = "[object Date]", uc = "[object Error]", lc = "[object Function]", cc = "[object Map]", fc = "[object Number]", dc = "[object Object]", pc = "[object RegExp]", gc = "[object Set]", hc = "[object String]", vc = "[object WeakMap]", bc = "[object ArrayBuffer]", yc = "[object DataView]", $c = "[object Float32Array]", _c = "[object Float64Array]", Tc = "[object Int8Array]", mc = "[object Int16Array]", Ec = "[object Int32Array]", Ac = "[object Uint8Array]", Oc = "[object Uint8ClampedArray]", Cc = "[object Uint16Array]", Nc = "[object Uint32Array]", O = {};
O[$c] = O[_c] = O[Tc] = O[mc] = O[Ec] = O[Ac] = O[Oc] = O[Cc] = O[Nc] = !0;
O[ac] = O[sc] = O[bc] = O[ic] = O[yc] = O[oc] = O[uc] = O[lc] = O[cc] = O[fc] = O[dc] = O[pc] = O[gc] = O[hc] = O[vc] = !1;
function wc(e) {
  return nc(e) && rc(e.length) && !!O[tc(e)];
}
var Pc = wc;
function Sc(e) {
  return function(t) {
    return e(t);
  };
}
var pt = Sc, Ce = { exports: {} };
Ce.exports;
(function(e, t) {
  var r = nn, n = t && !t.nodeType && t, a = n && !0 && e && !e.nodeType && e, s = a && a.exports === n, l = s && r.process, u = function() {
    try {
      var o = a && a.require && a.require("util").types;
      return o || l && l.binding && l.binding("util");
    } catch {
    }
  }();
  e.exports = u;
})(Ce, Ce.exports);
var gt = Ce.exports, xc = Pc, jc = pt, pr = gt, gr = pr && pr.isTypedArray, Ic = gr ? jc(gr) : xc, vn = Ic, Rc = Ll, Fc = gn, Lc = U, Mc = dt, Bc = Ql, Gc = vn, Vc = Object.prototype, Uc = Vc.hasOwnProperty;
function kc(e, t) {
  var r = Lc(e), n = !r && Fc(e), a = !r && !n && Mc(e), s = !r && !n && !a && Gc(e), l = r || n || a || s, u = l ? Rc(e.length, String) : [], o = u.length;
  for (var c in e)
    (t || Uc.call(e, c)) && !(l && // Safari 9 has enumerable `arguments.length` in strict mode.
    (c == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
    a && (c == "offset" || c == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
    s && (c == "buffer" || c == "byteLength" || c == "byteOffset") || // Skip index properties.
    Bc(c, o))) && u.push(c);
  return u;
}
var bn = kc, Kc = Object.prototype;
function Xc(e) {
  var t = e && e.constructor, r = typeof t == "function" && t.prototype || Kc;
  return e === r;
}
var ht = Xc;
function Wc(e, t) {
  return function(r) {
    return e(t(r));
  };
}
var yn = Wc, Yc = yn, qc = Yc(Object.keys, Object), Hc = qc, zc = ht, Jc = Hc, Zc = Object.prototype, Qc = Zc.hasOwnProperty;
function Dc(e) {
  if (!zc(e))
    return Jc(e);
  var t = [];
  for (var r in Object(e))
    Qc.call(e, r) && r != "constructor" && t.push(r);
  return t;
}
var ef = Dc, tf = sn, rf = hn;
function nf(e) {
  return e != null && rf(e.length) && !tf(e);
}
var $n = nf, af = bn, sf = ef, of = $n;
function uf(e) {
  return of(e) ? af(e) : sf(e);
}
var vt = uf, lf = fn, cf = ft, ff = vt;
function df(e) {
  return lf(e, ff, cf);
}
var _n = df, hr = _n, pf = 1, gf = Object.prototype, hf = gf.hasOwnProperty;
function vf(e, t, r, n, a, s) {
  var l = r & pf, u = hr(e), o = u.length, c = hr(t), d = c.length;
  if (o != d && !l)
    return !1;
  for (var g = o; g--; ) {
    var h = u[g];
    if (!(l ? h in t : hf.call(t, h)))
      return !1;
  }
  var v = s.get(e), T = s.get(t);
  if (v && T)
    return v == t && T == e;
  var w = !0;
  s.set(e, t), s.set(t, e);
  for (var C = l; ++g < o; ) {
    h = u[g];
    var P = e[h], N = t[h];
    if (n)
      var pe = l ? n(N, P, h, t, e, s) : n(P, N, h, e, t, s);
    if (!(pe === void 0 ? P === N || a(P, N, r, n, s) : pe)) {
      w = !1;
      break;
    }
    C || (C = h == "constructor");
  }
  if (w && !C) {
    var q = e.constructor, H = t.constructor;
    q != H && "constructor" in e && "constructor" in t && !(typeof q == "function" && q instanceof q && typeof H == "function" && H instanceof H) && (w = !1);
  }
  return s.delete(e), s.delete(t), w;
}
var bf = vf, yf = Y, $f = L, _f = yf($f, "DataView"), Tf = _f, mf = Y, Ef = L, Af = mf(Ef, "Promise"), Of = Af, Cf = Y, Nf = L, wf = Cf(Nf, "Set"), Pf = wf, Sf = Y, xf = L, jf = Sf(xf, "WeakMap"), If = jf, ze = Tf, Je = ut, Ze = Of, Qe = Pf, De = If, Tn = D, ne = on, vr = "[object Map]", Rf = "[object Object]", br = "[object Promise]", yr = "[object Set]", $r = "[object WeakMap]", _r = "[object DataView]", Ff = ne(ze), Lf = ne(Je), Mf = ne(Ze), Bf = ne(Qe), Gf = ne(De), X = Tn;
(ze && X(new ze(new ArrayBuffer(1))) != _r || Je && X(new Je()) != vr || Ze && X(Ze.resolve()) != br || Qe && X(new Qe()) != yr || De && X(new De()) != $r) && (X = function(e) {
  var t = Tn(e), r = t == Rf ? e.constructor : void 0, n = r ? ne(r) : "";
  if (n)
    switch (n) {
      case Ff:
        return _r;
      case Lf:
        return vr;
      case Mf:
        return br;
      case Bf:
        return yr;
      case Gf:
        return $r;
    }
  return t;
});
var Ie = X, Xe = un, Vf = ln, Uf = Tl, kf = bf, Tr = Ie, mr = U, Er = dt, Kf = vn, Xf = 1, Ar = "[object Arguments]", Or = "[object Array]", _e = "[object Object]", Wf = Object.prototype, Cr = Wf.hasOwnProperty;
function Yf(e, t, r, n, a, s) {
  var l = mr(e), u = mr(t), o = l ? Or : Tr(e), c = u ? Or : Tr(t);
  o = o == Ar ? _e : o, c = c == Ar ? _e : c;
  var d = o == _e, g = c == _e, h = o == c;
  if (h && Er(e)) {
    if (!Er(t))
      return !1;
    l = !0, d = !1;
  }
  if (h && !d)
    return s || (s = new Xe()), l || Kf(e) ? Vf(e, t, r, n, a, s) : Uf(e, t, o, r, n, a, s);
  if (!(r & Xf)) {
    var v = d && Cr.call(e, "__wrapped__"), T = g && Cr.call(t, "__wrapped__");
    if (v || T) {
      var w = v ? e.value() : e, C = T ? t.value() : t;
      return s || (s = new Xe()), a(w, C, r, n, s);
    }
  }
  return h ? (s || (s = new Xe()), kf(e, t, r, n, a, s)) : !1;
}
var qf = Yf, Hf = qf, Nr = k;
function mn(e, t, r, n, a) {
  return e === t ? !0 : e == null || t == null || !Nr(e) && !Nr(t) ? e !== e && t !== t : Hf(e, t, r, n, mn, a);
}
var zf = mn, Jf = zf;
function Zf(e, t) {
  return Jf(e, t);
}
var Qf = Zf;
const Df = /* @__PURE__ */ et(Qf);
function ed(e) {
  const { indentation: t, children: r, isInline: n } = e, { theme: a } = G();
  return /* @__PURE__ */ b.jsxs("div", { style: { color: a.cdataColor }, children: [
    /* @__PURE__ */ b.jsx("span", { children: `${t}<![CDATA[` }),
    r,
    /* @__PURE__ */ b.jsx("span", { children: `${n ? "" : t}]]>` })
  ] });
}
const td = (e) => /* @__PURE__ */ qt.createElement("svg", { width: "1em", height: "1em", viewBox: "0 0 24 24", ...e }, /* @__PURE__ */ qt.createElement("path", { d: "M9 17.898C9 18.972 10.2649 19.546 11.0731 18.8388L17.3838 13.3169C18.1806 12.6197 18.1806 11.3801 17.3838 10.6829L11.0731 5.16108C10.2649 4.45388 9 5.02785 9 6.1018V17.898Z" }));
function En(e) {
  const { collapsible: t, theme: r } = G(), { collapsed: n } = e;
  return t ? /* @__PURE__ */ b.jsx("span", { style: { position: "relative" }, children: /* @__PURE__ */ b.jsx("span", { style: { position: "absolute", right: "0", border: 0, padding: 0, background: "none" }, children: /* @__PURE__ */ b.jsx(
    td,
    {
      fill: r.separatorColor,
      style: { transform: `rotate(${n ? 0 : 90}deg)`, transition: "transform 0.2s" }
    }
  ) }) }) : null;
}
function rd(e) {
  return e == null;
}
var nd = rd;
const wr = /* @__PURE__ */ et(nd);
function An(e) {
  const { collapsible: t, initalCollapsedDepth: r } = G(), [n, a] = Hr(
    () => wr(r) || !t ? !1 : e >= r
  ), s = () => a((l) => !l);
  return zr(() => {
    a(
      wr(r) || !t ? !1 : e >= r
    );
  }, [r, e, t]), {
    collapsed: n,
    buttonProps: t ? {
      onClick: s,
      role: "button",
      style: { cursor: "pointer" }
    } : {}
  };
}
function ad(e) {
  const { indentation: t, children: r, isInline: n, level: a } = e, { theme: s } = G(), { collapsed: l, buttonProps: u } = An(a);
  return /* @__PURE__ */ b.jsxs("div", { style: { color: s.commentColor }, children: [
    /* @__PURE__ */ b.jsxs("span", { ...u, children: [
      /* @__PURE__ */ b.jsx("span", { children: t }),
      /* @__PURE__ */ b.jsx(En, { collapsed: l }),
      /* @__PURE__ */ b.jsx("span", { children: "<!-- " })
    ] }),
    !l && r,
    l && "...",
    /* @__PURE__ */ b.jsx("span", { children: `${n || l ? " " : t}-->` })
  ] });
}
function On(e) {
  const { attributes: t } = e, { theme: r } = G();
  return t ? /* @__PURE__ */ b.jsx("span", { children: Object.entries(t).map(([n, a]) => /* @__PURE__ */ b.jsxs("span", { children: [
    /* @__PURE__ */ b.jsx("span", { style: { color: r.attributeKeyColor }, children: ` ${n}` }),
    /* @__PURE__ */ b.jsx("span", { style: { color: r.separatorColor }, children: "=" }),
    /* @__PURE__ */ b.jsx("span", { style: { color: r.attributeValueColor }, children: `"${a}"` })
  ] }, `attribute-${n}`)) }) : null;
}
function sd(e) {
  const { indentation: t, tagKey: r, attributes: n } = e, { theme: a } = G();
  return /* @__PURE__ */ b.jsxs("div", { children: [
    /* @__PURE__ */ b.jsx("span", { style: { color: a.separatorColor }, children: `${t}<?` }),
    /* @__PURE__ */ b.jsx("span", { style: { color: a.tagColor }, children: `${r.slice(1)}` }),
    /* @__PURE__ */ b.jsx(On, { attributes: n }),
    /* @__PURE__ */ b.jsx("span", { style: { color: a.separatorColor }, children: "?>" })
  ] });
}
function id(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length, a = Array(n); ++r < n; )
    a[r] = t(e[r], r, e);
  return a;
}
var Cn = id;
function od(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length; ++r < n && t(e[r], r, e) !== !1; )
    ;
  return e;
}
var ud = od, ld = Y, cd = function() {
  try {
    var e = ld(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}(), Nn = cd, Pr = Nn;
function fd(e, t, r) {
  t == "__proto__" && Pr ? Pr(e, t, {
    configurable: !0,
    enumerable: !0,
    value: r,
    writable: !0
  }) : e[t] = r;
}
var wn = fd, dd = wn, pd = ot, gd = Object.prototype, hd = gd.hasOwnProperty;
function vd(e, t, r) {
  var n = e[t];
  (!(hd.call(e, t) && pd(n, r)) || r === void 0 && !(t in e)) && dd(e, t, r);
}
var Pn = vd, bd = Pn, yd = wn;
function $d(e, t, r, n) {
  var a = !r;
  r || (r = {});
  for (var s = -1, l = t.length; ++s < l; ) {
    var u = t[s], o = n ? n(r[u], e[u], u, r, e) : void 0;
    o === void 0 && (o = e[u]), a ? yd(r, u, o) : bd(r, u, o);
  }
  return r;
}
var de = $d, _d = de, Td = vt;
function md(e, t) {
  return e && _d(t, Td(t), e);
}
var Ed = md;
function Ad(e) {
  var t = [];
  if (e != null)
    for (var r in Object(e))
      t.push(r);
  return t;
}
var Od = Ad, Cd = fe, Nd = ht, wd = Od, Pd = Object.prototype, Sd = Pd.hasOwnProperty;
function xd(e) {
  if (!Cd(e))
    return wd(e);
  var t = Nd(e), r = [];
  for (var n in e)
    n == "constructor" && (t || !Sd.call(e, n)) || r.push(n);
  return r;
}
var jd = xd, Id = bn, Rd = jd, Fd = $n;
function Ld(e) {
  return Fd(e) ? Id(e, !0) : Rd(e);
}
var bt = Ld, Md = de, Bd = bt;
function Gd(e, t) {
  return e && Md(t, Bd(t), e);
}
var Vd = Gd, Ne = { exports: {} };
Ne.exports;
(function(e, t) {
  var r = L, n = t && !t.nodeType && t, a = n && !0 && e && !e.nodeType && e, s = a && a.exports === n, l = s ? r.Buffer : void 0, u = l ? l.allocUnsafe : void 0;
  function o(c, d) {
    if (d)
      return c.slice();
    var g = c.length, h = u ? u(g) : new c.constructor(g);
    return c.copy(h), h;
  }
  e.exports = o;
})(Ne, Ne.exports);
var Ud = Ne.exports;
function kd(e, t) {
  var r = -1, n = e.length;
  for (t || (t = Array(n)); ++r < n; )
    t[r] = e[r];
  return t;
}
var Kd = kd, Xd = de, Wd = ft;
function Yd(e, t) {
  return Xd(e, Wd(e), t);
}
var qd = Yd, Hd = yn, zd = Hd(Object.getPrototypeOf, Object), yt = zd, Jd = ct, Zd = yt, Qd = ft, Dd = dn, ep = Object.getOwnPropertySymbols, tp = ep ? function(e) {
  for (var t = []; e; )
    Jd(t, Qd(e)), e = Zd(e);
  return t;
} : Dd, Sn = tp, rp = de, np = Sn;
function ap(e, t) {
  return rp(e, np(e), t);
}
var sp = ap, ip = fn, op = Sn, up = bt;
function lp(e) {
  return ip(e, up, op);
}
var xn = lp, cp = Object.prototype, fp = cp.hasOwnProperty;
function dp(e) {
  var t = e.length, r = new e.constructor(t);
  return t && typeof e[0] == "string" && fp.call(e, "index") && (r.index = e.index, r.input = e.input), r;
}
var pp = dp, Sr = cn;
function gp(e) {
  var t = new e.constructor(e.byteLength);
  return new Sr(t).set(new Sr(e)), t;
}
var $t = gp, hp = $t;
function vp(e, t) {
  var r = t ? hp(e.buffer) : e.buffer;
  return new e.constructor(r, e.byteOffset, e.byteLength);
}
var bp = vp, yp = /\w*$/;
function $p(e) {
  var t = new e.constructor(e.source, yp.exec(e));
  return t.lastIndex = e.lastIndex, t;
}
var _p = $p, xr = Q, jr = xr ? xr.prototype : void 0, Ir = jr ? jr.valueOf : void 0;
function Tp(e) {
  return Ir ? Object(Ir.call(e)) : {};
}
var mp = Tp, Ep = $t;
function Ap(e, t) {
  var r = t ? Ep(e.buffer) : e.buffer;
  return new e.constructor(r, e.byteOffset, e.length);
}
var Op = Ap, Cp = $t, Np = bp, wp = _p, Pp = mp, Sp = Op, xp = "[object Boolean]", jp = "[object Date]", Ip = "[object Map]", Rp = "[object Number]", Fp = "[object RegExp]", Lp = "[object Set]", Mp = "[object String]", Bp = "[object Symbol]", Gp = "[object ArrayBuffer]", Vp = "[object DataView]", Up = "[object Float32Array]", kp = "[object Float64Array]", Kp = "[object Int8Array]", Xp = "[object Int16Array]", Wp = "[object Int32Array]", Yp = "[object Uint8Array]", qp = "[object Uint8ClampedArray]", Hp = "[object Uint16Array]", zp = "[object Uint32Array]";
function Jp(e, t, r) {
  var n = e.constructor;
  switch (t) {
    case Gp:
      return Cp(e);
    case xp:
    case jp:
      return new n(+e);
    case Vp:
      return Np(e, r);
    case Up:
    case kp:
    case Kp:
    case Xp:
    case Wp:
    case Yp:
    case qp:
    case Hp:
    case zp:
      return Sp(e, r);
    case Ip:
      return new n();
    case Rp:
    case Mp:
      return new n(e);
    case Fp:
      return wp(e);
    case Lp:
      return new n();
    case Bp:
      return Pp(e);
  }
}
var Zp = Jp, Qp = fe, Rr = Object.create, Dp = function() {
  function e() {
  }
  return function(t) {
    if (!Qp(t))
      return {};
    if (Rr)
      return Rr(t);
    e.prototype = t;
    var r = new e();
    return e.prototype = void 0, r;
  };
}(), eg = Dp, tg = eg, rg = yt, ng = ht;
function ag(e) {
  return typeof e.constructor == "function" && !ng(e) ? tg(rg(e)) : {};
}
var sg = ag, ig = Ie, og = k, ug = "[object Map]";
function lg(e) {
  return og(e) && ig(e) == ug;
}
var cg = lg, fg = cg, dg = pt, Fr = gt, Lr = Fr && Fr.isMap, pg = Lr ? dg(Lr) : fg, gg = pg, hg = Ie, vg = k, bg = "[object Set]";
function yg(e) {
  return vg(e) && hg(e) == bg;
}
var $g = yg, _g = $g, Tg = pt, Mr = gt, Br = Mr && Mr.isSet, mg = Br ? Tg(Br) : _g, Eg = mg, Ag = un, Og = ud, Cg = Pn, Ng = Ed, wg = Vd, Pg = Ud, Sg = Kd, xg = qd, jg = sp, Ig = _n, Rg = xn, Fg = Ie, Lg = pp, Mg = Zp, Bg = sg, Gg = U, Vg = dt, Ug = gg, kg = fe, Kg = Eg, Xg = vt, Wg = bt, Yg = 1, qg = 2, Hg = 4, jn = "[object Arguments]", zg = "[object Array]", Jg = "[object Boolean]", Zg = "[object Date]", Qg = "[object Error]", In = "[object Function]", Dg = "[object GeneratorFunction]", eh = "[object Map]", th = "[object Number]", Rn = "[object Object]", rh = "[object RegExp]", nh = "[object Set]", ah = "[object String]", sh = "[object Symbol]", ih = "[object WeakMap]", oh = "[object ArrayBuffer]", uh = "[object DataView]", lh = "[object Float32Array]", ch = "[object Float64Array]", fh = "[object Int8Array]", dh = "[object Int16Array]", ph = "[object Int32Array]", gh = "[object Uint8Array]", hh = "[object Uint8ClampedArray]", vh = "[object Uint16Array]", bh = "[object Uint32Array]", A = {};
A[jn] = A[zg] = A[oh] = A[uh] = A[Jg] = A[Zg] = A[lh] = A[ch] = A[fh] = A[dh] = A[ph] = A[eh] = A[th] = A[Rn] = A[rh] = A[nh] = A[ah] = A[sh] = A[gh] = A[hh] = A[vh] = A[bh] = !0;
A[Qg] = A[In] = A[ih] = !1;
function Te(e, t, r, n, a, s) {
  var l, u = t & Yg, o = t & qg, c = t & Hg;
  if (r && (l = a ? r(e, n, a, s) : r(e)), l !== void 0)
    return l;
  if (!kg(e))
    return e;
  var d = Gg(e);
  if (d) {
    if (l = Lg(e), !u)
      return Sg(e, l);
  } else {
    var g = Fg(e), h = g == In || g == Dg;
    if (Vg(e))
      return Pg(e, u);
    if (g == Rn || g == jn || h && !a) {
      if (l = o || h ? {} : Bg(e), !u)
        return o ? jg(e, wg(l, e)) : xg(e, Ng(l, e));
    } else {
      if (!A[g])
        return a ? e : {};
      l = Mg(e, g, u);
    }
  }
  s || (s = new Ag());
  var v = s.get(e);
  if (v)
    return v;
  s.set(e, l), Kg(e) ? e.forEach(function(C) {
    l.add(Te(C, t, r, C, e, s));
  }) : Ug(e) && e.forEach(function(C, P) {
    l.set(P, Te(C, t, r, P, e, s));
  });
  var T = c ? o ? Rg : Ig : o ? Wg : Xg, w = d ? void 0 : T(e);
  return Og(w || e, function(C, P) {
    w && (P = C, C = e[P]), Cg(l, P, Te(C, t, r, P, e, s));
  }), l;
}
var yh = Te, $h = D, _h = k, Th = "[object Symbol]";
function mh(e) {
  return typeof e == "symbol" || _h(e) && $h(e) == Th;
}
var _t = mh, Eh = U, Ah = _t, Oh = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, Ch = /^\w*$/;
function Nh(e, t) {
  if (Eh(e))
    return !1;
  var r = typeof e;
  return r == "number" || r == "symbol" || r == "boolean" || e == null || Ah(e) ? !0 : Ch.test(e) || !Oh.test(e) || t != null && e in Object(t);
}
var wh = Nh, Fn = lt, Ph = "Expected a function";
function Tt(e, t) {
  if (typeof e != "function" || t != null && typeof t != "function")
    throw new TypeError(Ph);
  var r = function() {
    var n = arguments, a = t ? t.apply(this, n) : n[0], s = r.cache;
    if (s.has(a))
      return s.get(a);
    var l = e.apply(this, n);
    return r.cache = s.set(a, l) || s, l;
  };
  return r.cache = new (Tt.Cache || Fn)(), r;
}
Tt.Cache = Fn;
var Sh = Tt, xh = Sh, jh = 500;
function Ih(e) {
  var t = xh(e, function(n) {
    return r.size === jh && r.clear(), n;
  }), r = t.cache;
  return t;
}
var Rh = Ih, Fh = Rh, Lh = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, Mh = /\\(\\)?/g, Bh = Fh(function(e) {
  var t = [];
  return e.charCodeAt(0) === 46 && t.push(""), e.replace(Lh, function(r, n, a, s) {
    t.push(a ? s.replace(Mh, "$1") : n || r);
  }), t;
}), Gh = Bh, Gr = Q, Vh = Cn, Uh = U, kh = _t, Kh = 1 / 0, Vr = Gr ? Gr.prototype : void 0, Ur = Vr ? Vr.toString : void 0;
function Ln(e) {
  if (typeof e == "string")
    return e;
  if (Uh(e))
    return Vh(e, Ln) + "";
  if (kh(e))
    return Ur ? Ur.call(e) : "";
  var t = e + "";
  return t == "0" && 1 / e == -Kh ? "-0" : t;
}
var Xh = Ln, Wh = Xh;
function Yh(e) {
  return e == null ? "" : Wh(e);
}
var qh = Yh, Hh = U, zh = wh, Jh = Gh, Zh = qh;
function Qh(e, t) {
  return Hh(e) ? e : zh(e, t) ? [e] : Jh(Zh(e));
}
var mt = Qh;
function Dh(e) {
  var t = e == null ? 0 : e.length;
  return t ? e[t - 1] : void 0;
}
var ev = Dh, tv = _t, rv = 1 / 0;
function nv(e) {
  if (typeof e == "string" || tv(e))
    return e;
  var t = e + "";
  return t == "0" && 1 / e == -rv ? "-0" : t;
}
var Mn = nv, av = mt, sv = Mn;
function iv(e, t) {
  t = av(t, e);
  for (var r = 0, n = t.length; e != null && r < n; )
    e = e[sv(t[r++])];
  return r && r == n ? e : void 0;
}
var ov = iv;
function uv(e, t, r) {
  var n = -1, a = e.length;
  t < 0 && (t = -t > a ? 0 : a + t), r = r > a ? a : r, r < 0 && (r += a), a = t > r ? 0 : r - t >>> 0, t >>>= 0;
  for (var s = Array(a); ++n < a; )
    s[n] = e[n + t];
  return s;
}
var lv = uv, cv = ov, fv = lv;
function dv(e, t) {
  return t.length < 2 ? e : cv(e, fv(t, 0, -1));
}
var pv = dv, gv = mt, hv = ev, vv = pv, bv = Mn;
function yv(e, t) {
  return t = gv(t, e), e = vv(e, t), e == null || delete e[bv(hv(t))];
}
var $v = yv, _v = D, Tv = yt, mv = k, Ev = "[object Object]", Av = Function.prototype, Ov = Object.prototype, Bn = Av.toString, Cv = Ov.hasOwnProperty, Nv = Bn.call(Object);
function wv(e) {
  if (!mv(e) || _v(e) != Ev)
    return !1;
  var t = Tv(e);
  if (t === null)
    return !0;
  var r = Cv.call(t, "constructor") && t.constructor;
  return typeof r == "function" && r instanceof r && Bn.call(r) == Nv;
}
var Pv = wv, Sv = Pv;
function xv(e) {
  return Sv(e) ? void 0 : e;
}
var jv = xv, kr = Q, Iv = gn, Rv = U, Kr = kr ? kr.isConcatSpreadable : void 0;
function Fv(e) {
  return Rv(e) || Iv(e) || !!(Kr && e && e[Kr]);
}
var Lv = Fv, Mv = ct, Bv = Lv;
function Gn(e, t, r, n, a) {
  var s = -1, l = e.length;
  for (r || (r = Bv), a || (a = []); ++s < l; ) {
    var u = e[s];
    t > 0 && r(u) ? t > 1 ? Gn(u, t - 1, r, n, a) : Mv(a, u) : n || (a[a.length] = u);
  }
  return a;
}
var Gv = Gn, Vv = Gv;
function Uv(e) {
  var t = e == null ? 0 : e.length;
  return t ? Vv(e, 1) : [];
}
var kv = Uv;
function Kv(e, t, r) {
  switch (r.length) {
    case 0:
      return e.call(t);
    case 1:
      return e.call(t, r[0]);
    case 2:
      return e.call(t, r[0], r[1]);
    case 3:
      return e.call(t, r[0], r[1], r[2]);
  }
  return e.apply(t, r);
}
var Xv = Kv, Wv = Xv, Xr = Math.max;
function Yv(e, t, r) {
  return t = Xr(t === void 0 ? e.length - 1 : t, 0), function() {
    for (var n = arguments, a = -1, s = Xr(n.length - t, 0), l = Array(s); ++a < s; )
      l[a] = n[t + a];
    a = -1;
    for (var u = Array(t + 1); ++a < t; )
      u[a] = n[a];
    return u[t] = r(l), Wv(e, this, u);
  };
}
var qv = Yv;
function Hv(e) {
  return function() {
    return e;
  };
}
var zv = Hv;
function Jv(e) {
  return e;
}
var Zv = Jv, Qv = zv, Wr = Nn, Dv = Zv, eb = Wr ? function(e, t) {
  return Wr(e, "toString", {
    configurable: !0,
    enumerable: !1,
    value: Qv(t),
    writable: !0
  });
} : Dv, tb = eb, rb = 800, nb = 16, ab = Date.now;
function sb(e) {
  var t = 0, r = 0;
  return function() {
    var n = ab(), a = nb - (n - r);
    if (r = n, a > 0) {
      if (++t >= rb)
        return arguments[0];
    } else
      t = 0;
    return e.apply(void 0, arguments);
  };
}
var ib = sb, ob = tb, ub = ib, lb = ub(ob), cb = lb, fb = kv, db = qv, pb = cb;
function gb(e) {
  return pb(db(e, void 0, fb), e + "");
}
var hb = gb, vb = Cn, bb = yh, yb = $v, $b = mt, _b = de, Tb = jv, mb = hb, Eb = xn, Ab = 1, Ob = 2, Cb = 4, Nb = mb(function(e, t) {
  var r = {};
  if (e == null)
    return r;
  var n = !1;
  t = vb(t, function(s) {
    return s = $b(s, e), n || (n = s.length > 1), s;
  }), _b(e, Eb(e), r), n && (r = bb(r, Ab | Ob | Cb, Tb));
  for (var a = t.length; a--; )
    yb(r, t[a]);
  return r;
}), wb = Nb;
const Pb = /* @__PURE__ */ et(wb);
function Sb(e, t) {
  return new Array(t * e + 1).join(" ");
}
function Vn(e) {
  return typeof e == "string" && e.includes(`
`);
}
function xb(e) {
  return typeof e == "string" ? !1 : e.length === 1 && Ee in e[0] && !Vn(e[0][Ee]);
}
function jb(e) {
  switch (e) {
    case Ee:
    case rt:
    case tt:
      return e;
    default:
      return e.startsWith("?") ? Jr : Ta;
  }
}
function Ib(e) {
  const t = e[Jt], r = Pb(e, Jt), [[n, a]] = Object.entries(r), s = jb(n);
  return { attributes: t, tagKey: n, subElements: a, type: s };
}
function Rb(e) {
  return e ? Object.keys(e).length > 0 : !1;
}
function Fb(e) {
  const { indentation: t, tagKey: r, attributes: n, children: a, isInline: s, hasChildren: l, level: u } = e, { collapsed: o, buttonProps: c } = An(u), { theme: d } = G();
  return /* @__PURE__ */ b.jsxs("div", { children: [
    /* @__PURE__ */ b.jsxs("span", { ...c, children: [
      /* @__PURE__ */ b.jsx("span", { children: t }),
      /* @__PURE__ */ b.jsx(En, { collapsed: o }),
      /* @__PURE__ */ b.jsx("span", { style: { color: d.separatorColor }, children: "<" }),
      /* @__PURE__ */ b.jsx("span", { style: { color: d.tagColor }, children: `${r}` }),
      !o && /* @__PURE__ */ b.jsx(On, { attributes: n }),
      o && Rb(n) && " ...",
      /* @__PURE__ */ b.jsx("span", { style: { color: d.separatorColor }, children: l ? ">" : " />" })
    ] }),
    l && /* @__PURE__ */ b.jsxs(b.Fragment, { children: [
      !o && a,
      o && "...",
      /* @__PURE__ */ b.jsx("span", { style: { color: d.separatorColor }, children: `${s || o ? "" : t}</` }),
      /* @__PURE__ */ b.jsx("span", { style: { color: d.tagColor }, children: `${r}` }),
      /* @__PURE__ */ b.jsx("span", { style: { color: d.separatorColor }, children: ">" })
    ] })
  ] });
}
function Lb(e) {
  const { hasSiblings: t, text: r, indentation: n, isText: a } = e, { theme: s } = G(), l = a ? { color: s.textColor } : void 0;
  return Vn(r) || t ? /* @__PURE__ */ b.jsx("div", { style: l, children: r.split(`
`).filter((u) => !!u.trim()).map((u, o) => /* @__PURE__ */ b.jsx("div", { children: `${n}${u.trim()}` }, `${o}`)) }) : /* @__PURE__ */ b.jsx("span", { style: l, children: r });
}
function me(e) {
  const { elements: t, level: r = 0, isText: n = !0 } = e, { indentSize: a } = G();
  return !Array.isArray(t) || t.length === 0 ? null : /* @__PURE__ */ b.jsx(b.Fragment, { children: t.map((s, l) => {
    const { tagKey: u, attributes: o, subElements: c, type: d } = Ib(s), g = t.length > 1, h = Sb(a, r), v = `${r}-${l}`, T = xb(c);
    switch (d) {
      case Ee:
        return /* @__PURE__ */ b.jsx(
          Lb,
          {
            text: c,
            indentation: h,
            hasSiblings: g,
            isText: n
          },
          v
        );
      case tt:
        return /* @__PURE__ */ b.jsx(ad, { isInline: T, indentation: h, level: r, children: /* @__PURE__ */ b.jsx(me, { elements: c, level: r + 1, isText: !1 }) }, v);
      case rt:
        return /* @__PURE__ */ b.jsx(ed, { indentation: h, isInline: T, children: /* @__PURE__ */ b.jsx(me, { elements: c, level: r + 1, isText: !1 }) }, v);
      case Jr:
        return /* @__PURE__ */ b.jsx(
          sd,
          {
            indentation: h,
            tagKey: u,
            attributes: o
          },
          v
        );
      default:
        return /* @__PURE__ */ b.jsx(
          Fb,
          {
            indentation: h,
            tagKey: u,
            attributes: o,
            isInline: T,
            hasChildren: c.length > 0,
            level: r,
            children: /* @__PURE__ */ b.jsx(me, { elements: c, level: r + 1 })
          },
          v
        );
    }
  }) });
}
function Mb() {
  return /* @__PURE__ */ b.jsx("div", { children: "Invalid XML!" });
}
function Ub(e) {
  const {
    theme: t,
    xml: r,
    collapsible: n = !1,
    indentSize: a = 2,
    invalidXml: s,
    initalCollapsedDepth: l
  } = e, [u, o] = Hr(() => ({ ...Ye, ...t })), { json: c, valid: d } = Gs(r), g = qr(
    () => ({ theme: u, collapsible: n, indentSize: a, initalCollapsedDepth: l }),
    [u, n, a, l]
  );
  return zr(() => {
    o((h) => {
      const v = { ...Ye, ...t };
      return Df(v, h) ? h : v;
    });
  }, [t]), d ? /* @__PURE__ */ b.jsx(Zr.Provider, { value: g, children: /* @__PURE__ */ b.jsx(
    "div",
    {
      className: "rxv-container",
      style: { whiteSpace: "pre-wrap", fontFamily: u.fontFamily, overflowWrap: "break-word" },
      children: /* @__PURE__ */ b.jsx(me, { elements: c })
    }
  ) }) : s || /* @__PURE__ */ b.jsx(Mb, {});
}
export {
  Ub as default
};
